{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "c4c9c1b4-6404-439a-a33e-fc3d1c3b668e",
   "metadata": {},
   "source": [
    "# DESeq2\n",
    "# Phase 1 vs. Phase 1\n",
    "\n",
    "focusing on pairwise comparisons involving oysters that experienced only **one** phase 1 (one exposure)\n",
    "\n",
    "## 0. load libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "2738b967-b031-47d2-b595-b2ea3eeeb3db",
   "metadata": {},
   "outputs": [],
   "source": [
    "library(tidyverse)\n",
    "library(DESeq2)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cf2cdd9b-45ac-4041-94f9-aa04eb0fb748",
   "metadata": {},
   "source": [
    "## 1. read counts matrix and meta data\n",
    "\n",
    "### counts matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "bbf503d8-7f17-4f12-90d9-3b824f61712e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 120</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>B1_B1_O01</th><th scope=col>B1_Nu_O03</th><th scope=col>B1_W5_O50</th><th scope=col>B2_B5_O51</th><th scope=col>B2_C4_O40</th><th scope=col>B2_Nu_O12</th><th scope=col>B3_B4_O41</th><th scope=col>B3_C3_O30</th><th scope=col>B3_C6_O66</th><th scope=col>B3_H4_O41</th><th scope=col>⋯</th><th scope=col>W5_C4_G45</th><th scope=col>W5_H4_G46</th><th scope=col>W5_W2_G22</th><th scope=col>W6_B3_G35</th><th scope=col>W6_B4_G48</th><th scope=col>W6_H6_G71</th><th scope=col>W6_Nu_G41</th><th scope=col>W6_Nu_G45</th><th scope=col>W6_W3_G36</th><th scope=col>W6_W4_G48</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>⋯</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>LOC111116054</th><td>  2</td><td>  1</td><td>  6</td><td> 10</td><td>  9</td><td> 16</td><td>  0</td><td> 10</td><td> 22</td><td>  9</td><td>⋯</td><td>  9</td><td>  2</td><td>  8</td><td>  5</td><td>  2</td><td> 19</td><td>  1</td><td>  2</td><td>  1</td><td>  0</td></tr>\n",
       "\t<tr><th scope=row>LOC111126949</th><td>885</td><td>652</td><td>477</td><td>654</td><td>586</td><td>523</td><td>392</td><td>357</td><td>496</td><td>540</td><td>⋯</td><td>407</td><td>740</td><td>707</td><td>406</td><td>418</td><td>424</td><td>492</td><td>330</td><td>281</td><td>599</td></tr>\n",
       "\t<tr><th scope=row>LOC111110729</th><td> 64</td><td>209</td><td> 93</td><td> 63</td><td>100</td><td>177</td><td> 76</td><td> 98</td><td>235</td><td>181</td><td>⋯</td><td> 70</td><td>149</td><td>121</td><td>115</td><td>126</td><td>108</td><td>118</td><td>115</td><td>127</td><td>213</td></tr>\n",
       "\t<tr><th scope=row>LOC111112434</th><td> 11</td><td>  7</td><td>  2</td><td>  0</td><td>  2</td><td>  2</td><td> 12</td><td> 15</td><td> 10</td><td> 36</td><td>⋯</td><td> 11</td><td>  6</td><td>  4</td><td>  0</td><td>  2</td><td> 16</td><td> 22</td><td>  0</td><td>  8</td><td>  0</td></tr>\n",
       "\t<tr><th scope=row>LOC111120752</th><td>360</td><td>586</td><td>336</td><td>426</td><td>351</td><td>417</td><td>236</td><td>278</td><td>761</td><td>373</td><td>⋯</td><td>359</td><td>345</td><td>438</td><td>278</td><td>287</td><td>416</td><td>621</td><td>251</td><td>333</td><td>430</td></tr>\n",
       "\t<tr><th scope=row>LOC111128944</th><td>  0</td><td>  0</td><td>  0</td><td>  0</td><td>  0</td><td>  0</td><td>  0</td><td>  0</td><td>  0</td><td>  0</td><td>⋯</td><td>  0</td><td>  0</td><td>  0</td><td>  0</td><td>  0</td><td>  0</td><td>  0</td><td>  1</td><td>  0</td><td>  0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 120\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & B1\\_B1\\_O01 & B1\\_Nu\\_O03 & B1\\_W5\\_O50 & B2\\_B5\\_O51 & B2\\_C4\\_O40 & B2\\_Nu\\_O12 & B3\\_B4\\_O41 & B3\\_C3\\_O30 & B3\\_C6\\_O66 & B3\\_H4\\_O41 & ⋯ & W5\\_C4\\_G45 & W5\\_H4\\_G46 & W5\\_W2\\_G22 & W6\\_B3\\_G35 & W6\\_B4\\_G48 & W6\\_H6\\_G71 & W6\\_Nu\\_G41 & W6\\_Nu\\_G45 & W6\\_W3\\_G36 & W6\\_W4\\_G48\\\\\n",
       "  & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & ⋯ & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\tLOC111116054 &   2 &   1 &   6 &  10 &   9 &  16 &   0 &  10 &  22 &   9 & ⋯ &   9 &   2 &   8 &   5 &   2 &  19 &   1 &   2 &   1 &   0\\\\\n",
       "\tLOC111126949 & 885 & 652 & 477 & 654 & 586 & 523 & 392 & 357 & 496 & 540 & ⋯ & 407 & 740 & 707 & 406 & 418 & 424 & 492 & 330 & 281 & 599\\\\\n",
       "\tLOC111110729 &  64 & 209 &  93 &  63 & 100 & 177 &  76 &  98 & 235 & 181 & ⋯ &  70 & 149 & 121 & 115 & 126 & 108 & 118 & 115 & 127 & 213\\\\\n",
       "\tLOC111112434 &  11 &   7 &   2 &   0 &   2 &   2 &  12 &  15 &  10 &  36 & ⋯ &  11 &   6 &   4 &   0 &   2 &  16 &  22 &   0 &   8 &   0\\\\\n",
       "\tLOC111120752 & 360 & 586 & 336 & 426 & 351 & 417 & 236 & 278 & 761 & 373 & ⋯ & 359 & 345 & 438 & 278 & 287 & 416 & 621 & 251 & 333 & 430\\\\\n",
       "\tLOC111128944 &   0 &   0 &   0 &   0 &   0 &   0 &   0 &   0 &   0 &   0 & ⋯ &   0 &   0 &   0 &   0 &   0 &   0 &   0 &   1 &   0 &   0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 120\n",
       "\n",
       "| <!--/--> | B1_B1_O01 &lt;int&gt; | B1_Nu_O03 &lt;int&gt; | B1_W5_O50 &lt;int&gt; | B2_B5_O51 &lt;int&gt; | B2_C4_O40 &lt;int&gt; | B2_Nu_O12 &lt;int&gt; | B3_B4_O41 &lt;int&gt; | B3_C3_O30 &lt;int&gt; | B3_C6_O66 &lt;int&gt; | B3_H4_O41 &lt;int&gt; | ⋯ ⋯ | W5_C4_G45 &lt;int&gt; | W5_H4_G46 &lt;int&gt; | W5_W2_G22 &lt;int&gt; | W6_B3_G35 &lt;int&gt; | W6_B4_G48 &lt;int&gt; | W6_H6_G71 &lt;int&gt; | W6_Nu_G41 &lt;int&gt; | W6_Nu_G45 &lt;int&gt; | W6_W3_G36 &lt;int&gt; | W6_W4_G48 &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| LOC111116054 |   2 |   1 |   6 |  10 |   9 |  16 |   0 |  10 |  22 |   9 | ⋯ |   9 |   2 |   8 |   5 |   2 |  19 |   1 |   2 |   1 |   0 |\n",
       "| LOC111126949 | 885 | 652 | 477 | 654 | 586 | 523 | 392 | 357 | 496 | 540 | ⋯ | 407 | 740 | 707 | 406 | 418 | 424 | 492 | 330 | 281 | 599 |\n",
       "| LOC111110729 |  64 | 209 |  93 |  63 | 100 | 177 |  76 |  98 | 235 | 181 | ⋯ |  70 | 149 | 121 | 115 | 126 | 108 | 118 | 115 | 127 | 213 |\n",
       "| LOC111112434 |  11 |   7 |   2 |   0 |   2 |   2 |  12 |  15 |  10 |  36 | ⋯ |  11 |   6 |   4 |   0 |   2 |  16 |  22 |   0 |   8 |   0 |\n",
       "| LOC111120752 | 360 | 586 | 336 | 426 | 351 | 417 | 236 | 278 | 761 | 373 | ⋯ | 359 | 345 | 438 | 278 | 287 | 416 | 621 | 251 | 333 | 430 |\n",
       "| LOC111128944 |   0 |   0 |   0 |   0 |   0 |   0 |   0 |   0 |   0 |   0 | ⋯ |   0 |   0 |   0 |   0 |   0 |   0 |   0 |   1 |   0 |   0 |\n",
       "\n"
      ],
      "text/plain": [
       "             B1_B1_O01 B1_Nu_O03 B1_W5_O50 B2_B5_O51 B2_C4_O40 B2_Nu_O12\n",
       "LOC111116054   2         1         6        10         9        16      \n",
       "LOC111126949 885       652       477       654       586       523      \n",
       "LOC111110729  64       209        93        63       100       177      \n",
       "LOC111112434  11         7         2         0         2         2      \n",
       "LOC111120752 360       586       336       426       351       417      \n",
       "LOC111128944   0         0         0         0         0         0      \n",
       "             B3_B4_O41 B3_C3_O30 B3_C6_O66 B3_H4_O41 ⋯ W5_C4_G45 W5_H4_G46\n",
       "LOC111116054   0        10        22         9       ⋯   9         2      \n",
       "LOC111126949 392       357       496       540       ⋯ 407       740      \n",
       "LOC111110729  76        98       235       181       ⋯  70       149      \n",
       "LOC111112434  12        15        10        36       ⋯  11         6      \n",
       "LOC111120752 236       278       761       373       ⋯ 359       345      \n",
       "LOC111128944   0         0         0         0       ⋯   0         0      \n",
       "             W5_W2_G22 W6_B3_G35 W6_B4_G48 W6_H6_G71 W6_Nu_G41 W6_Nu_G45\n",
       "LOC111116054   8         5         2        19         1         2      \n",
       "LOC111126949 707       406       418       424       492       330      \n",
       "LOC111110729 121       115       126       108       118       115      \n",
       "LOC111112434   4         0         2        16        22         0      \n",
       "LOC111120752 438       278       287       416       621       251      \n",
       "LOC111128944   0         0         0         0         0         1      \n",
       "             W6_W3_G36 W6_W4_G48\n",
       "LOC111116054   1         0      \n",
       "LOC111126949 281       599      \n",
       "LOC111110729 127       213      \n",
       "LOC111112434   8         0      \n",
       "LOC111120752 333       430      \n",
       "LOC111128944   0         0      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# counts matrix\n",
    "counts <- read.csv('/work/pi_sarah_gignouxwolfsohn_uml_edu/julia_mcdonough_student_uml_edu/ce24_rnaseq/featureCounts/featureCounts_matrix.csv')\n",
    "\n",
    "# set gene_ID as row name\n",
    "rownames(counts) <- counts$Gene_ID\n",
    "\n",
    "# remove gene id and length\n",
    "counts <- counts %>%\n",
    "select(-Gene_ID, -Length)\n",
    "\n",
    "head(counts)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "abefc129-3603-49d3-9c1f-34447c4abe75",
   "metadata": {},
   "source": [
    "### meta data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "61ea51e0-7de5-466b-9154-476d88d902d4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 6</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Sample</th><th scope=col>Phase1_Treatment</th><th scope=col>Phase1_TankRep</th><th scope=col>Phase2_Treatment</th><th scope=col>Phase2_TankRep</th><th scope=col>Tag</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>B1_B1_O01</td><td>B</td><td>1</td><td>B </td><td> 1</td><td>O01</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>B1_Nu_O03</td><td>B</td><td>1</td><td>Nu</td><td>NA</td><td>O03</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>B1_W5_O50</td><td>B</td><td>1</td><td>W </td><td> 5</td><td>O50</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>B2_B5_O51</td><td>B</td><td>2</td><td>B </td><td> 5</td><td>O51</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>B2_C4_O40</td><td>B</td><td>2</td><td>C </td><td> 4</td><td>O40</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>B2_Nu_O12</td><td>B</td><td>2</td><td>Nu</td><td>NA</td><td>O12</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 6\n",
       "\\begin{tabular}{r|llllll}\n",
       "  & Sample & Phase1\\_Treatment & Phase1\\_TankRep & Phase2\\_Treatment & Phase2\\_TankRep & Tag\\\\\n",
       "  & <chr> & <chr> & <dbl> & <chr> & <dbl> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & B1\\_B1\\_O01 & B & 1 & B  &  1 & O01\\\\\n",
       "\t2 & B1\\_Nu\\_O03 & B & 1 & Nu & NA & O03\\\\\n",
       "\t3 & B1\\_W5\\_O50 & B & 1 & W  &  5 & O50\\\\\n",
       "\t4 & B2\\_B5\\_O51 & B & 2 & B  &  5 & O51\\\\\n",
       "\t5 & B2\\_C4\\_O40 & B & 2 & C  &  4 & O40\\\\\n",
       "\t6 & B2\\_Nu\\_O12 & B & 2 & Nu & NA & O12\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 6\n",
       "\n",
       "| <!--/--> | Sample &lt;chr&gt; | Phase1_Treatment &lt;chr&gt; | Phase1_TankRep &lt;dbl&gt; | Phase2_Treatment &lt;chr&gt; | Phase2_TankRep &lt;dbl&gt; | Tag &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| 1 | B1_B1_O01 | B | 1 | B  |  1 | O01 |\n",
       "| 2 | B1_Nu_O03 | B | 1 | Nu | NA | O03 |\n",
       "| 3 | B1_W5_O50 | B | 1 | W  |  5 | O50 |\n",
       "| 4 | B2_B5_O51 | B | 2 | B  |  5 | O51 |\n",
       "| 5 | B2_C4_O40 | B | 2 | C  |  4 | O40 |\n",
       "| 6 | B2_Nu_O12 | B | 2 | Nu | NA | O12 |\n",
       "\n"
      ],
      "text/plain": [
       "  Sample    Phase1_Treatment Phase1_TankRep Phase2_Treatment Phase2_TankRep Tag\n",
       "1 B1_B1_O01 B                1              B                 1             O01\n",
       "2 B1_Nu_O03 B                1              Nu               NA             O03\n",
       "3 B1_W5_O50 B                1              W                 5             O50\n",
       "4 B2_B5_O51 B                2              B                 5             O51\n",
       "5 B2_C4_O40 B                2              C                 4             O40\n",
       "6 B2_Nu_O12 B                2              Nu               NA             O12"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# create df with col names from counts matrix\n",
    "meta <- as.data.frame(colnames(counts2))\n",
    "\n",
    "colnames(meta) <- 'Sample'\n",
    "\n",
    "# Separate into columns\n",
    "meta <- meta %>%\n",
    "   # Keep original SampleID\n",
    "  mutate(OGSample = Sample) %>%\n",
    "  # Split by '_'\n",
    "  separate(Sample, into = c(\"Phase1\", \"Phase2\", \"Tag\"), sep = \"_\") %>%\n",
    "  mutate(\n",
    "    Phase1_Treatment = substr(Phase1, 1, 1),\n",
    "    Phase1_TankRep    = as.numeric(substr(Phase1, 2, nchar(Phase1))),\n",
    "    Phase2_Treatment = gsub(\"[0-9]\", \"\", Phase2),\n",
    "    Phase2_TankRep       = as.numeric(gsub(\"[A-Za-z]\", \"\", Phase2))\n",
    "  ) %>%\n",
    "  select(OGSample, Phase1_Treatment, Phase1_TankRep, Phase2_Treatment, Phase2_TankRep, Tag)\n",
    "\n",
    "colnames(meta)[1] <- 'Sample'\n",
    "\n",
    "head(meta)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "a066c4ce-5d08-498d-8311-d764b34e7852",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 9</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Sample</th><th scope=col>Phase1_treatment</th><th scope=col>Phase1_temp</th><th scope=col>Phase1_DO</th><th scope=col>Phase1_TankRep</th><th scope=col>Phase2_treatment</th><th scope=col>Phase2_temp</th><th scope=col>Phase2_DO</th><th scope=col>Phase2_TankRep</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>B1_B1_O01</td><td>both</td><td>warm</td><td>hypoxic</td><td>1</td><td>both   </td><td>warm   </td><td>hypoxic </td><td> 1</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>B1_Nu_O03</td><td>both</td><td>warm</td><td>hypoxic</td><td>1</td><td>NA     </td><td>NA     </td><td>NA      </td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>B1_W5_O50</td><td>both</td><td>warm</td><td>hypoxic</td><td>1</td><td>warm   </td><td>warm   </td><td>normoxic</td><td> 5</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>B2_B5_O51</td><td>both</td><td>warm</td><td>hypoxic</td><td>2</td><td>both   </td><td>warm   </td><td>hypoxic </td><td> 5</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>B2_C4_O40</td><td>both</td><td>warm</td><td>hypoxic</td><td>2</td><td>control</td><td>ambient</td><td>normoxic</td><td> 4</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>B2_Nu_O12</td><td>both</td><td>warm</td><td>hypoxic</td><td>2</td><td>NA     </td><td>NA     </td><td>NA      </td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 9\n",
       "\\begin{tabular}{r|lllllllll}\n",
       "  & Sample & Phase1\\_treatment & Phase1\\_temp & Phase1\\_DO & Phase1\\_TankRep & Phase2\\_treatment & Phase2\\_temp & Phase2\\_DO & Phase2\\_TankRep\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <dbl> & <chr> & <chr> & <chr> & <dbl>\\\\\n",
       "\\hline\n",
       "\t1 & B1\\_B1\\_O01 & both & warm & hypoxic & 1 & both    & warm    & hypoxic  &  1\\\\\n",
       "\t2 & B1\\_Nu\\_O03 & both & warm & hypoxic & 1 & NA      & NA      & NA       & NA\\\\\n",
       "\t3 & B1\\_W5\\_O50 & both & warm & hypoxic & 1 & warm    & warm    & normoxic &  5\\\\\n",
       "\t4 & B2\\_B5\\_O51 & both & warm & hypoxic & 2 & both    & warm    & hypoxic  &  5\\\\\n",
       "\t5 & B2\\_C4\\_O40 & both & warm & hypoxic & 2 & control & ambient & normoxic &  4\\\\\n",
       "\t6 & B2\\_Nu\\_O12 & both & warm & hypoxic & 2 & NA      & NA      & NA       & NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 9\n",
       "\n",
       "| <!--/--> | Sample &lt;chr&gt; | Phase1_treatment &lt;chr&gt; | Phase1_temp &lt;chr&gt; | Phase1_DO &lt;chr&gt; | Phase1_TankRep &lt;dbl&gt; | Phase2_treatment &lt;chr&gt; | Phase2_temp &lt;chr&gt; | Phase2_DO &lt;chr&gt; | Phase2_TankRep &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | B1_B1_O01 | both | warm | hypoxic | 1 | both    | warm    | hypoxic  |  1 |\n",
       "| 2 | B1_Nu_O03 | both | warm | hypoxic | 1 | NA      | NA      | NA       | NA |\n",
       "| 3 | B1_W5_O50 | both | warm | hypoxic | 1 | warm    | warm    | normoxic |  5 |\n",
       "| 4 | B2_B5_O51 | both | warm | hypoxic | 2 | both    | warm    | hypoxic  |  5 |\n",
       "| 5 | B2_C4_O40 | both | warm | hypoxic | 2 | control | ambient | normoxic |  4 |\n",
       "| 6 | B2_Nu_O12 | both | warm | hypoxic | 2 | NA      | NA      | NA       | NA |\n",
       "\n"
      ],
      "text/plain": [
       "  Sample    Phase1_treatment Phase1_temp Phase1_DO Phase1_TankRep\n",
       "1 B1_B1_O01 both             warm        hypoxic   1             \n",
       "2 B1_Nu_O03 both             warm        hypoxic   1             \n",
       "3 B1_W5_O50 both             warm        hypoxic   1             \n",
       "4 B2_B5_O51 both             warm        hypoxic   2             \n",
       "5 B2_C4_O40 both             warm        hypoxic   2             \n",
       "6 B2_Nu_O12 both             warm        hypoxic   2             \n",
       "  Phase2_treatment Phase2_temp Phase2_DO Phase2_TankRep\n",
       "1 both             warm        hypoxic    1            \n",
       "2 NA               NA          NA        NA            \n",
       "3 warm             warm        normoxic   5            \n",
       "4 both             warm        hypoxic    5            \n",
       "5 control          ambient     normoxic   4            \n",
       "6 NA               NA          NA        NA            "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# adding treatment info\n",
    "meta2 <- meta %>%\n",
    "mutate(\n",
    "    ## phase 1 treatments\n",
    "    # overall\n",
    "    Phase1_treatment = case_when(\n",
    "        Phase1_Treatment == 'B' ~ 'both',\n",
    "        Phase1_Treatment == 'H' ~ 'hypoxic',\n",
    "        Phase1_Treatment == 'W' ~ 'warm',\n",
    "        Phase1_Treatment == 'C' ~ 'control',\n",
    "    TRUE ~ NA_character_),\n",
    "    \n",
    "    # temp\n",
    "    Phase1_temp = case_when(\n",
    "        Phase1_Treatment == 'B' ~ 'warm',\n",
    "        Phase1_Treatment == 'H' ~ 'ambient',\n",
    "        Phase1_Treatment == 'W' ~ 'warm',\n",
    "        Phase1_Treatment == 'C' ~ 'ambient',\n",
    "    TRUE ~ NA_character_),\n",
    "\n",
    "    # dissolved oxygen\n",
    "    Phase1_DO = case_when(\n",
    "        Phase1_Treatment == 'B' ~ 'hypoxic',\n",
    "        Phase1_Treatment == 'H' ~ 'hypoxic',\n",
    "        Phase1_Treatment == 'W' ~ 'normoxic',\n",
    "        Phase1_Treatment == 'C' ~ 'normoxic',\n",
    "    TRUE ~ NA_character_),\n",
    "    \n",
    "    ## phase 2 treatments\n",
    "    # overall\n",
    "    Phase2_treatment = case_when(\n",
    "        Phase2_Treatment == 'B' ~ 'both',\n",
    "        Phase2_Treatment == 'H' ~ 'hypoxic',\n",
    "        Phase2_Treatment == 'W' ~ 'warm',\n",
    "        Phase2_Treatment == 'C' ~ 'control',\n",
    "    TRUE ~ NA_character_),\n",
    "    \n",
    "    # temp\n",
    "    Phase2_temp = case_when(\n",
    "        Phase2_Treatment == 'B' ~ 'warm',\n",
    "        Phase2_Treatment == 'H' ~ 'ambient',\n",
    "        Phase2_Treatment == 'W' ~ 'warm',\n",
    "        Phase2_Treatment == 'C' ~ 'ambient',\n",
    "    TRUE ~ NA_character_),\n",
    "\n",
    "    # dissolved oxygen\n",
    "    Phase2_DO = case_when(\n",
    "        Phase2_Treatment == 'B' ~ 'hypoxic',\n",
    "        Phase2_Treatment == 'H' ~ 'hypoxic',\n",
    "        Phase2_Treatment == 'W' ~ 'normoxic',\n",
    "        Phase2_Treatment == 'C' ~ 'normoxic',\n",
    "    TRUE ~ NA_character_)) %>%\n",
    "select(Sample, Phase1_treatment, Phase1_temp, Phase1_DO, Phase1_TankRep, Phase2_treatment, Phase2_temp, Phase2_DO, Phase2_TankRep)\n",
    "\n",
    "head(meta2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "050b285f-81fd-414a-b262-098a9bfa3d31",
   "metadata": {},
   "outputs": [],
   "source": [
    "write.csv(meta2, '/project/pi_sarah_gignouxwolfsohn_uml_edu/julia/CE_2024/CE24_RNA-seq/sample_metaData.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6f675085-4085-4171-8ea7-ed106974abfe",
   "metadata": {},
   "source": [
    "for this analysis, only looking at Phase 1 samples, so pulling those out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e841cd50-f62c-4fde-a51b-8da2713099ad",
   "metadata": {},
   "outputs": [],
   "source": [
    "# if skipped above code:\n",
    "meta2 <- read.csv('/project/pi_sarah_gignouxwolfsohn_uml_edu/julia/CE_2024/CE24_RNA-seq/sample_metaData.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "fdfa19a3-85a7-47da-9bfd-d719d2f0c823",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 23 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Phase1_treatment</th><th scope=col>Phase1_temp</th><th scope=col>Phase1_DO</th><th scope=col>Phase1_TankRep</th><th scope=col>Phase2_treatment</th><th scope=col>Phase2_temp</th><th scope=col>Phase2_DO</th><th scope=col>Phase2_TankRep</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>B1_Nu_O03</th><td>both   </td><td>warm   </td><td>hypoxic </td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>B2_Nu_O12</th><td>both   </td><td>warm   </td><td>hypoxic </td><td>2</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>B4_Nu_O32</th><td>both   </td><td>warm   </td><td>hypoxic </td><td>4</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>B5_Nu_O36</th><td>both   </td><td>warm   </td><td>hypoxic </td><td>5</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>B6_Nu_O47</th><td>both   </td><td>warm   </td><td>hypoxic </td><td>6</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>C1_Nu_W01</th><td>control</td><td>ambient</td><td>normoxic</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>C1_Nu_W05</th><td>control</td><td>ambient</td><td>normoxic</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>C2_Nu_W15</th><td>control</td><td>ambient</td><td>normoxic</td><td>2</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>C3_Nu_W21</th><td>control</td><td>ambient</td><td>normoxic</td><td>3</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>C4_Nu_W29</th><td>control</td><td>ambient</td><td>normoxic</td><td>4</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>C6_Nu_W42</th><td>control</td><td>ambient</td><td>normoxic</td><td>6</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>H1_Nu_B08</th><td>hypoxic</td><td>ambient</td><td>hypoxic </td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>H2_Nu_B14</th><td>hypoxic</td><td>ambient</td><td>hypoxic </td><td>2</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>H3_Nu_B18</th><td>hypoxic</td><td>ambient</td><td>hypoxic </td><td>3</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>H4_Nu_B28</th><td>hypoxic</td><td>ambient</td><td>hypoxic </td><td>4</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>H5_Nu_B35</th><td>hypoxic</td><td>ambient</td><td>hypoxic </td><td>5</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>H6_Nu_B45</th><td>hypoxic</td><td>ambient</td><td>hypoxic </td><td>6</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>W1_Nu_G02</th><td>warm   </td><td>warm   </td><td>normoxic</td><td>1</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>W2_Nu_G15</th><td>warm   </td><td>warm   </td><td>normoxic</td><td>2</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>W3_Nu_G21</th><td>warm   </td><td>warm   </td><td>normoxic</td><td>3</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>W4_Nu_G27</th><td>warm   </td><td>warm   </td><td>normoxic</td><td>4</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>W6_Nu_G41</th><td>warm   </td><td>warm   </td><td>normoxic</td><td>6</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>W6_Nu_G45</th><td>warm   </td><td>warm   </td><td>normoxic</td><td>6</td><td>NA</td><td>NA</td><td>NA</td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 23 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & Phase1\\_treatment & Phase1\\_temp & Phase1\\_DO & Phase1\\_TankRep & Phase2\\_treatment & Phase2\\_temp & Phase2\\_DO & Phase2\\_TankRep\\\\\n",
       "  & <chr> & <chr> & <chr> & <dbl> & <chr> & <chr> & <chr> & <dbl>\\\\\n",
       "\\hline\n",
       "\tB1\\_Nu\\_O03 & both    & warm    & hypoxic  & 1 & NA & NA & NA & NA\\\\\n",
       "\tB2\\_Nu\\_O12 & both    & warm    & hypoxic  & 2 & NA & NA & NA & NA\\\\\n",
       "\tB4\\_Nu\\_O32 & both    & warm    & hypoxic  & 4 & NA & NA & NA & NA\\\\\n",
       "\tB5\\_Nu\\_O36 & both    & warm    & hypoxic  & 5 & NA & NA & NA & NA\\\\\n",
       "\tB6\\_Nu\\_O47 & both    & warm    & hypoxic  & 6 & NA & NA & NA & NA\\\\\n",
       "\tC1\\_Nu\\_W01 & control & ambient & normoxic & 1 & NA & NA & NA & NA\\\\\n",
       "\tC1\\_Nu\\_W05 & control & ambient & normoxic & 1 & NA & NA & NA & NA\\\\\n",
       "\tC2\\_Nu\\_W15 & control & ambient & normoxic & 2 & NA & NA & NA & NA\\\\\n",
       "\tC3\\_Nu\\_W21 & control & ambient & normoxic & 3 & NA & NA & NA & NA\\\\\n",
       "\tC4\\_Nu\\_W29 & control & ambient & normoxic & 4 & NA & NA & NA & NA\\\\\n",
       "\tC6\\_Nu\\_W42 & control & ambient & normoxic & 6 & NA & NA & NA & NA\\\\\n",
       "\tH1\\_Nu\\_B08 & hypoxic & ambient & hypoxic  & 1 & NA & NA & NA & NA\\\\\n",
       "\tH2\\_Nu\\_B14 & hypoxic & ambient & hypoxic  & 2 & NA & NA & NA & NA\\\\\n",
       "\tH3\\_Nu\\_B18 & hypoxic & ambient & hypoxic  & 3 & NA & NA & NA & NA\\\\\n",
       "\tH4\\_Nu\\_B28 & hypoxic & ambient & hypoxic  & 4 & NA & NA & NA & NA\\\\\n",
       "\tH5\\_Nu\\_B35 & hypoxic & ambient & hypoxic  & 5 & NA & NA & NA & NA\\\\\n",
       "\tH6\\_Nu\\_B45 & hypoxic & ambient & hypoxic  & 6 & NA & NA & NA & NA\\\\\n",
       "\tW1\\_Nu\\_G02 & warm    & warm    & normoxic & 1 & NA & NA & NA & NA\\\\\n",
       "\tW2\\_Nu\\_G15 & warm    & warm    & normoxic & 2 & NA & NA & NA & NA\\\\\n",
       "\tW3\\_Nu\\_G21 & warm    & warm    & normoxic & 3 & NA & NA & NA & NA\\\\\n",
       "\tW4\\_Nu\\_G27 & warm    & warm    & normoxic & 4 & NA & NA & NA & NA\\\\\n",
       "\tW6\\_Nu\\_G41 & warm    & warm    & normoxic & 6 & NA & NA & NA & NA\\\\\n",
       "\tW6\\_Nu\\_G45 & warm    & warm    & normoxic & 6 & NA & NA & NA & NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 23 × 8\n",
       "\n",
       "| <!--/--> | Phase1_treatment &lt;chr&gt; | Phase1_temp &lt;chr&gt; | Phase1_DO &lt;chr&gt; | Phase1_TankRep &lt;dbl&gt; | Phase2_treatment &lt;chr&gt; | Phase2_temp &lt;chr&gt; | Phase2_DO &lt;chr&gt; | Phase2_TankRep &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| B1_Nu_O03 | both    | warm    | hypoxic  | 1 | NA | NA | NA | NA |\n",
       "| B2_Nu_O12 | both    | warm    | hypoxic  | 2 | NA | NA | NA | NA |\n",
       "| B4_Nu_O32 | both    | warm    | hypoxic  | 4 | NA | NA | NA | NA |\n",
       "| B5_Nu_O36 | both    | warm    | hypoxic  | 5 | NA | NA | NA | NA |\n",
       "| B6_Nu_O47 | both    | warm    | hypoxic  | 6 | NA | NA | NA | NA |\n",
       "| C1_Nu_W01 | control | ambient | normoxic | 1 | NA | NA | NA | NA |\n",
       "| C1_Nu_W05 | control | ambient | normoxic | 1 | NA | NA | NA | NA |\n",
       "| C2_Nu_W15 | control | ambient | normoxic | 2 | NA | NA | NA | NA |\n",
       "| C3_Nu_W21 | control | ambient | normoxic | 3 | NA | NA | NA | NA |\n",
       "| C4_Nu_W29 | control | ambient | normoxic | 4 | NA | NA | NA | NA |\n",
       "| C6_Nu_W42 | control | ambient | normoxic | 6 | NA | NA | NA | NA |\n",
       "| H1_Nu_B08 | hypoxic | ambient | hypoxic  | 1 | NA | NA | NA | NA |\n",
       "| H2_Nu_B14 | hypoxic | ambient | hypoxic  | 2 | NA | NA | NA | NA |\n",
       "| H3_Nu_B18 | hypoxic | ambient | hypoxic  | 3 | NA | NA | NA | NA |\n",
       "| H4_Nu_B28 | hypoxic | ambient | hypoxic  | 4 | NA | NA | NA | NA |\n",
       "| H5_Nu_B35 | hypoxic | ambient | hypoxic  | 5 | NA | NA | NA | NA |\n",
       "| H6_Nu_B45 | hypoxic | ambient | hypoxic  | 6 | NA | NA | NA | NA |\n",
       "| W1_Nu_G02 | warm    | warm    | normoxic | 1 | NA | NA | NA | NA |\n",
       "| W2_Nu_G15 | warm    | warm    | normoxic | 2 | NA | NA | NA | NA |\n",
       "| W3_Nu_G21 | warm    | warm    | normoxic | 3 | NA | NA | NA | NA |\n",
       "| W4_Nu_G27 | warm    | warm    | normoxic | 4 | NA | NA | NA | NA |\n",
       "| W6_Nu_G41 | warm    | warm    | normoxic | 6 | NA | NA | NA | NA |\n",
       "| W6_Nu_G45 | warm    | warm    | normoxic | 6 | NA | NA | NA | NA |\n",
       "\n"
      ],
      "text/plain": [
       "          Phase1_treatment Phase1_temp Phase1_DO Phase1_TankRep\n",
       "B1_Nu_O03 both             warm        hypoxic   1             \n",
       "B2_Nu_O12 both             warm        hypoxic   2             \n",
       "B4_Nu_O32 both             warm        hypoxic   4             \n",
       "B5_Nu_O36 both             warm        hypoxic   5             \n",
       "B6_Nu_O47 both             warm        hypoxic   6             \n",
       "C1_Nu_W01 control          ambient     normoxic  1             \n",
       "C1_Nu_W05 control          ambient     normoxic  1             \n",
       "C2_Nu_W15 control          ambient     normoxic  2             \n",
       "C3_Nu_W21 control          ambient     normoxic  3             \n",
       "C4_Nu_W29 control          ambient     normoxic  4             \n",
       "C6_Nu_W42 control          ambient     normoxic  6             \n",
       "H1_Nu_B08 hypoxic          ambient     hypoxic   1             \n",
       "H2_Nu_B14 hypoxic          ambient     hypoxic   2             \n",
       "H3_Nu_B18 hypoxic          ambient     hypoxic   3             \n",
       "H4_Nu_B28 hypoxic          ambient     hypoxic   4             \n",
       "H5_Nu_B35 hypoxic          ambient     hypoxic   5             \n",
       "H6_Nu_B45 hypoxic          ambient     hypoxic   6             \n",
       "W1_Nu_G02 warm             warm        normoxic  1             \n",
       "W2_Nu_G15 warm             warm        normoxic  2             \n",
       "W3_Nu_G21 warm             warm        normoxic  3             \n",
       "W4_Nu_G27 warm             warm        normoxic  4             \n",
       "W6_Nu_G41 warm             warm        normoxic  6             \n",
       "W6_Nu_G45 warm             warm        normoxic  6             \n",
       "          Phase2_treatment Phase2_temp Phase2_DO Phase2_TankRep\n",
       "B1_Nu_O03 NA               NA          NA        NA            \n",
       "B2_Nu_O12 NA               NA          NA        NA            \n",
       "B4_Nu_O32 NA               NA          NA        NA            \n",
       "B5_Nu_O36 NA               NA          NA        NA            \n",
       "B6_Nu_O47 NA               NA          NA        NA            \n",
       "C1_Nu_W01 NA               NA          NA        NA            \n",
       "C1_Nu_W05 NA               NA          NA        NA            \n",
       "C2_Nu_W15 NA               NA          NA        NA            \n",
       "C3_Nu_W21 NA               NA          NA        NA            \n",
       "C4_Nu_W29 NA               NA          NA        NA            \n",
       "C6_Nu_W42 NA               NA          NA        NA            \n",
       "H1_Nu_B08 NA               NA          NA        NA            \n",
       "H2_Nu_B14 NA               NA          NA        NA            \n",
       "H3_Nu_B18 NA               NA          NA        NA            \n",
       "H4_Nu_B28 NA               NA          NA        NA            \n",
       "H5_Nu_B35 NA               NA          NA        NA            \n",
       "H6_Nu_B45 NA               NA          NA        NA            \n",
       "W1_Nu_G02 NA               NA          NA        NA            \n",
       "W2_Nu_G15 NA               NA          NA        NA            \n",
       "W3_Nu_G21 NA               NA          NA        NA            \n",
       "W4_Nu_G27 NA               NA          NA        NA            \n",
       "W6_Nu_G41 NA               NA          NA        NA            \n",
       "W6_Nu_G45 NA               NA          NA        NA            "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# selecting only phase 1 sample\n",
    "p1_meta <- meta2 %>%\n",
    "filter(grepl(\"Nu\", Sample)) %>%\n",
    "filter(Sample != 'B3_Nu_O24') # this sample failed all QC steps, removing from analysis\n",
    "\n",
    "# setting samples as rownames and deleting col from df\n",
    "rownames(p1_meta) <- p1_meta$Sample\n",
    "p1_meta <- p1_meta[, !colnames(p1_meta) %in% \"Sample\"]\n",
    "\n",
    "p1_meta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "431158f0-6fb5-4da0-866a-2f430ca10621",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 23</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>B1_Nu_O03</th><th scope=col>B2_Nu_O12</th><th scope=col>B4_Nu_O32</th><th scope=col>B5_Nu_O36</th><th scope=col>B6_Nu_O47</th><th scope=col>C1_Nu_W01</th><th scope=col>C1_Nu_W05</th><th scope=col>C2_Nu_W15</th><th scope=col>C3_Nu_W21</th><th scope=col>C4_Nu_W29</th><th scope=col>⋯</th><th scope=col>H3_Nu_B18</th><th scope=col>H4_Nu_B28</th><th scope=col>H5_Nu_B35</th><th scope=col>H6_Nu_B45</th><th scope=col>W1_Nu_G02</th><th scope=col>W2_Nu_G15</th><th scope=col>W3_Nu_G21</th><th scope=col>W4_Nu_G27</th><th scope=col>W6_Nu_G41</th><th scope=col>W6_Nu_G45</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>⋯</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>LOC111116054</th><td>  1</td><td> 16</td><td> 13</td><td>  11</td><td>  1</td><td> 16</td><td>   2</td><td>  1</td><td>  2</td><td> 10</td><td>⋯</td><td> 16</td><td> 29</td><td> 15</td><td>  0</td><td>   3</td><td>  23</td><td> 12</td><td>  9</td><td>  1</td><td>  2</td></tr>\n",
       "\t<tr><th scope=row>LOC111126949</th><td>652</td><td>523</td><td>444</td><td>1438</td><td>609</td><td>525</td><td>1076</td><td>541</td><td>566</td><td>613</td><td>⋯</td><td>451</td><td>608</td><td>745</td><td>394</td><td>1362</td><td>1235</td><td>270</td><td>407</td><td>492</td><td>330</td></tr>\n",
       "\t<tr><th scope=row>LOC111110729</th><td>209</td><td>177</td><td>207</td><td> 154</td><td>184</td><td>148</td><td> 144</td><td>161</td><td>155</td><td>259</td><td>⋯</td><td>136</td><td>151</td><td>221</td><td>143</td><td> 231</td><td>  64</td><td> 91</td><td>140</td><td>118</td><td>115</td></tr>\n",
       "\t<tr><th scope=row>LOC111112434</th><td>  7</td><td>  2</td><td> 10</td><td>  19</td><td> 18</td><td> 11</td><td>  17</td><td> 21</td><td> 18</td><td> 18</td><td>⋯</td><td>  6</td><td>  6</td><td> 25</td><td>  5</td><td>  24</td><td>   0</td><td> 10</td><td>  7</td><td> 22</td><td>  0</td></tr>\n",
       "\t<tr><th scope=row>LOC111120752</th><td>586</td><td>417</td><td>388</td><td> 202</td><td>330</td><td>848</td><td> 236</td><td>156</td><td>699</td><td>712</td><td>⋯</td><td>278</td><td>406</td><td>282</td><td>230</td><td> 430</td><td> 458</td><td>929</td><td>433</td><td>621</td><td>251</td></tr>\n",
       "\t<tr><th scope=row>LOC111128944</th><td>  0</td><td>  0</td><td>  0</td><td>   0</td><td>  0</td><td>  0</td><td>   2</td><td>  4</td><td>  0</td><td>  6</td><td>⋯</td><td> 14</td><td>  2</td><td>  0</td><td>  0</td><td>   2</td><td>   0</td><td>  4</td><td>  0</td><td>  0</td><td>  1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 23\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & B1\\_Nu\\_O03 & B2\\_Nu\\_O12 & B4\\_Nu\\_O32 & B5\\_Nu\\_O36 & B6\\_Nu\\_O47 & C1\\_Nu\\_W01 & C1\\_Nu\\_W05 & C2\\_Nu\\_W15 & C3\\_Nu\\_W21 & C4\\_Nu\\_W29 & ⋯ & H3\\_Nu\\_B18 & H4\\_Nu\\_B28 & H5\\_Nu\\_B35 & H6\\_Nu\\_B45 & W1\\_Nu\\_G02 & W2\\_Nu\\_G15 & W3\\_Nu\\_G21 & W4\\_Nu\\_G27 & W6\\_Nu\\_G41 & W6\\_Nu\\_G45\\\\\n",
       "  & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & ⋯ & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\tLOC111116054 &   1 &  16 &  13 &   11 &   1 &  16 &    2 &   1 &   2 &  10 & ⋯ &  16 &  29 &  15 &   0 &    3 &   23 &  12 &   9 &   1 &   2\\\\\n",
       "\tLOC111126949 & 652 & 523 & 444 & 1438 & 609 & 525 & 1076 & 541 & 566 & 613 & ⋯ & 451 & 608 & 745 & 394 & 1362 & 1235 & 270 & 407 & 492 & 330\\\\\n",
       "\tLOC111110729 & 209 & 177 & 207 &  154 & 184 & 148 &  144 & 161 & 155 & 259 & ⋯ & 136 & 151 & 221 & 143 &  231 &   64 &  91 & 140 & 118 & 115\\\\\n",
       "\tLOC111112434 &   7 &   2 &  10 &   19 &  18 &  11 &   17 &  21 &  18 &  18 & ⋯ &   6 &   6 &  25 &   5 &   24 &    0 &  10 &   7 &  22 &   0\\\\\n",
       "\tLOC111120752 & 586 & 417 & 388 &  202 & 330 & 848 &  236 & 156 & 699 & 712 & ⋯ & 278 & 406 & 282 & 230 &  430 &  458 & 929 & 433 & 621 & 251\\\\\n",
       "\tLOC111128944 &   0 &   0 &   0 &    0 &   0 &   0 &    2 &   4 &   0 &   6 & ⋯ &  14 &   2 &   0 &   0 &    2 &    0 &   4 &   0 &   0 &   1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 23\n",
       "\n",
       "| <!--/--> | B1_Nu_O03 &lt;int&gt; | B2_Nu_O12 &lt;int&gt; | B4_Nu_O32 &lt;int&gt; | B5_Nu_O36 &lt;int&gt; | B6_Nu_O47 &lt;int&gt; | C1_Nu_W01 &lt;int&gt; | C1_Nu_W05 &lt;int&gt; | C2_Nu_W15 &lt;int&gt; | C3_Nu_W21 &lt;int&gt; | C4_Nu_W29 &lt;int&gt; | ⋯ ⋯ | H3_Nu_B18 &lt;int&gt; | H4_Nu_B28 &lt;int&gt; | H5_Nu_B35 &lt;int&gt; | H6_Nu_B45 &lt;int&gt; | W1_Nu_G02 &lt;int&gt; | W2_Nu_G15 &lt;int&gt; | W3_Nu_G21 &lt;int&gt; | W4_Nu_G27 &lt;int&gt; | W6_Nu_G41 &lt;int&gt; | W6_Nu_G45 &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| LOC111116054 |   1 |  16 |  13 |   11 |   1 |  16 |    2 |   1 |   2 |  10 | ⋯ |  16 |  29 |  15 |   0 |    3 |   23 |  12 |   9 |   1 |   2 |\n",
       "| LOC111126949 | 652 | 523 | 444 | 1438 | 609 | 525 | 1076 | 541 | 566 | 613 | ⋯ | 451 | 608 | 745 | 394 | 1362 | 1235 | 270 | 407 | 492 | 330 |\n",
       "| LOC111110729 | 209 | 177 | 207 |  154 | 184 | 148 |  144 | 161 | 155 | 259 | ⋯ | 136 | 151 | 221 | 143 |  231 |   64 |  91 | 140 | 118 | 115 |\n",
       "| LOC111112434 |   7 |   2 |  10 |   19 |  18 |  11 |   17 |  21 |  18 |  18 | ⋯ |   6 |   6 |  25 |   5 |   24 |    0 |  10 |   7 |  22 |   0 |\n",
       "| LOC111120752 | 586 | 417 | 388 |  202 | 330 | 848 |  236 | 156 | 699 | 712 | ⋯ | 278 | 406 | 282 | 230 |  430 |  458 | 929 | 433 | 621 | 251 |\n",
       "| LOC111128944 |   0 |   0 |   0 |    0 |   0 |   0 |    2 |   4 |   0 |   6 | ⋯ |  14 |   2 |   0 |   0 |    2 |    0 |   4 |   0 |   0 |   1 |\n",
       "\n"
      ],
      "text/plain": [
       "             B1_Nu_O03 B2_Nu_O12 B4_Nu_O32 B5_Nu_O36 B6_Nu_O47 C1_Nu_W01\n",
       "LOC111116054   1        16        13         11        1        16      \n",
       "LOC111126949 652       523       444       1438      609       525      \n",
       "LOC111110729 209       177       207        154      184       148      \n",
       "LOC111112434   7         2        10         19       18        11      \n",
       "LOC111120752 586       417       388        202      330       848      \n",
       "LOC111128944   0         0         0          0        0         0      \n",
       "             C1_Nu_W05 C2_Nu_W15 C3_Nu_W21 C4_Nu_W29 ⋯ H3_Nu_B18 H4_Nu_B28\n",
       "LOC111116054    2        1         2        10       ⋯  16        29      \n",
       "LOC111126949 1076      541       566       613       ⋯ 451       608      \n",
       "LOC111110729  144      161       155       259       ⋯ 136       151      \n",
       "LOC111112434   17       21        18        18       ⋯   6         6      \n",
       "LOC111120752  236      156       699       712       ⋯ 278       406      \n",
       "LOC111128944    2        4         0         6       ⋯  14         2      \n",
       "             H5_Nu_B35 H6_Nu_B45 W1_Nu_G02 W2_Nu_G15 W3_Nu_G21 W4_Nu_G27\n",
       "LOC111116054  15         0          3        23       12         9      \n",
       "LOC111126949 745       394       1362      1235      270       407      \n",
       "LOC111110729 221       143        231        64       91       140      \n",
       "LOC111112434  25         5         24         0       10         7      \n",
       "LOC111120752 282       230        430       458      929       433      \n",
       "LOC111128944   0         0          2         0        4         0      \n",
       "             W6_Nu_G41 W6_Nu_G45\n",
       "LOC111116054   1         2      \n",
       "LOC111126949 492       330      \n",
       "LOC111110729 118       115      \n",
       "LOC111112434  22         0      \n",
       "LOC111120752 621       251      \n",
       "LOC111128944   0         1      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# pulling out only phase 1 counts\n",
    "p1_counts <- counts[,colnames(counts) %in% rownames(p1_meta)]\n",
    "head(p1_counts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "f5b4dd07-86c4-4c95-a0d4-919e3845cc95",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "TRUE"
      ],
      "text/latex": [
       "TRUE"
      ],
      "text/markdown": [
       "TRUE"
      ],
      "text/plain": [
       "[1] TRUE"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# setting the same order for counts matrix and meta data\n",
    "p1_counts <- p1_counts[,rownames(p1_meta)]\n",
    "all(rownames(p1_meta) == colnames(p1_counts))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "689025d2-6dd0-47ce-8529-f37a21375c86",
   "metadata": {},
   "source": [
    "## 2. DESeq2\n",
    "analyzing differential gene expression with [`DESeq2`](https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html)\n",
    "\n",
    "setting up contrasts (telling DESeq what is the control treatment)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "id": "b45d02ac-50b1-4f1f-8b6d-d0f57266fe9c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# overall phase 1 treatment\n",
    "p1_meta$Phase1_treatment <- factor(p1_meta$Phase1_treatment,\n",
    "                                   levels = c('control', 'both', 'hypoxic', 'warm'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "a7aedcbf-b5fe-452a-a229-92ba45de58a4",
   "metadata": {},
   "outputs": [],
   "source": [
    "# phase 1 tempature\n",
    "p1_meta$Phase1_temp <- factor(p1_meta$Phase1_temp,\n",
    "                                   levels = c('ambient', 'warm'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "id": "937d44e8-8196-49bf-a6d9-2bc432725206",
   "metadata": {},
   "outputs": [],
   "source": [
    "# phase 1 DO\n",
    "p1_meta$Phase1_DO <- factor(p1_meta$Phase1_DO,\n",
    "                                   levels = c('normoxic', 'hypoxic'))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3a2c0c39-f245-4f4b-94f1-080c4174f3f1",
   "metadata": {},
   "source": [
    "### phase 1 overall treatments"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "id": "ee56b1d4-93a4-461f-bc07-f7e6bdf21176",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "class: DESeqDataSet \n",
       "dim: 38838 23 \n",
       "metadata(1): version\n",
       "assays(1): counts\n",
       "rownames(38838): LOC111116054 LOC111126949 ... ND1 ND4L\n",
       "rowData names(0):\n",
       "colnames(23): B1_Nu_O03 B2_Nu_O12 ... W6_Nu_G41 W6_Nu_G45\n",
       "colData names(8): Phase1_treatment Phase1_temp ... Phase2_DO\n",
       "  Phase2_TankRep"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "estimating size factors\n",
      "\n",
      "estimating dispersions\n",
      "\n",
      "gene-wise dispersion estimates\n",
      "\n",
      "mean-dispersion relationship\n",
      "\n",
      "final dispersion estimates\n",
      "\n",
      "fitting model and testing\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'Intercept'</li><li>'Phase1_DO_hypoxic_vs_normoxic'</li><li>'Phase1_temp_warm_vs_ambient'</li><li>'Phase1_DOhypoxic.Phase1_tempwarm'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'Intercept'\n",
       "\\item 'Phase1\\_DO\\_hypoxic\\_vs\\_normoxic'\n",
       "\\item 'Phase1\\_temp\\_warm\\_vs\\_ambient'\n",
       "\\item 'Phase1\\_DOhypoxic.Phase1\\_tempwarm'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'Intercept'\n",
       "2. 'Phase1_DO_hypoxic_vs_normoxic'\n",
       "3. 'Phase1_temp_warm_vs_ambient'\n",
       "4. 'Phase1_DOhypoxic.Phase1_tempwarm'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"Intercept\"                        \"Phase1_DO_hypoxic_vs_normoxic\"   \n",
       "[3] \"Phase1_temp_warm_vs_ambient\"      \"Phase1_DOhypoxic.Phase1_tempwarm\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# creating DESeq2 object\n",
    "dds <- DESeqDataSetFromMatrix(countData = p1_counts,\n",
    "                              colData = p1_meta,\n",
    "                              design = ~ Phase1_DO + Phase1_temp + Phase1_DO:Phase1_temp)\n",
    "dds\n",
    "\n",
    "# running DESeq\n",
    "dds <- DESeq(dds)\n",
    "\n",
    "# list coefficients\n",
    "resultsNames(dds)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "deb8958a-e031-4d71-bc61-6e6a2f073957",
   "metadata": {},
   "source": [
    "### "
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R [conda env:.conda-r-env]",
   "language": "R",
   "name": "conda-env-.conda-r-env-r"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.3.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
