{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "de0aeb25-a373-4169-ac92-0d3247220b25",
   "metadata": {},
   "source": [
    "# WGCNA\n",
    "Weighted gene correlation network for analysis\n",
    "\n",
    "following [this tutorial](https://bioinformaticsworkbook.org/dataAnalysis/RNA-Seq/RNA-SeqIntro/wgcna.html)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ecef4742-bee8-47be-813f-156f219932e2",
   "metadata": {},
   "source": [
    "## 0. load libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "5e3fca6a-0da5-4c13-89ab-08d607cd3e5d",
   "metadata": {},
   "outputs": [],
   "source": [
    "library(tidyverse)\n",
    "library(DESeq2)\n",
    "library(WGCNA)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "05d24d2b-b9f6-4506-aec5-07f5ea0b169b",
   "metadata": {},
   "source": [
    "## 1. read CSVs\n",
    "set-up is similar to DESeq - read in counts matrix and meta data, then run DESeq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "568a2385-8498-4d36-b04e-e8c60da9b87c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>38838</li><li>119</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 38838\n",
       "\\item 119\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 38838\n",
       "2. 119\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 38838   119"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 119</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>B1_B1_O01</th><th scope=col>B1_Nu_O03</th><th scope=col>B1_W5_O50</th><th scope=col>B2_B5_O51</th><th scope=col>B2_C4_O40</th><th scope=col>B2_Nu_O12</th><th scope=col>B3_B4_O41</th><th scope=col>B3_C3_O30</th><th scope=col>B3_C6_O66</th><th scope=col>B3_H4_O41</th><th scope=col>⋯</th><th scope=col>W5_C4_G45</th><th scope=col>W5_H4_G46</th><th scope=col>W5_W2_G22</th><th scope=col>W6_B3_G35</th><th scope=col>W6_B4_G48</th><th scope=col>W6_H6_G71</th><th scope=col>W6_Nu_G41</th><th scope=col>W6_Nu_G45</th><th scope=col>W6_W3_G36</th><th scope=col>W6_W4_G48</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>⋯</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>LOC111116054</th><td>  2</td><td>  1</td><td>  6</td><td> 10</td><td>  9</td><td> 16</td><td>  0</td><td> 10</td><td> 22</td><td>  9</td><td>⋯</td><td>  9</td><td>  2</td><td>  8</td><td>  5</td><td>  2</td><td> 19</td><td>  1</td><td>  2</td><td>  1</td><td>  0</td></tr>\n",
       "\t<tr><th scope=row>LOC111126949</th><td>885</td><td>652</td><td>477</td><td>654</td><td>586</td><td>523</td><td>392</td><td>357</td><td>496</td><td>540</td><td>⋯</td><td>407</td><td>740</td><td>707</td><td>406</td><td>418</td><td>424</td><td>492</td><td>330</td><td>281</td><td>599</td></tr>\n",
       "\t<tr><th scope=row>LOC111110729</th><td> 64</td><td>209</td><td> 93</td><td> 63</td><td>100</td><td>177</td><td> 76</td><td> 98</td><td>235</td><td>181</td><td>⋯</td><td> 70</td><td>149</td><td>121</td><td>115</td><td>126</td><td>108</td><td>118</td><td>115</td><td>127</td><td>213</td></tr>\n",
       "\t<tr><th scope=row>LOC111112434</th><td> 11</td><td>  7</td><td>  2</td><td>  0</td><td>  2</td><td>  2</td><td> 12</td><td> 15</td><td> 10</td><td> 36</td><td>⋯</td><td> 11</td><td>  6</td><td>  4</td><td>  0</td><td>  2</td><td> 16</td><td> 22</td><td>  0</td><td>  8</td><td>  0</td></tr>\n",
       "\t<tr><th scope=row>LOC111120752</th><td>360</td><td>586</td><td>336</td><td>426</td><td>351</td><td>417</td><td>236</td><td>278</td><td>761</td><td>373</td><td>⋯</td><td>359</td><td>345</td><td>438</td><td>278</td><td>287</td><td>416</td><td>621</td><td>251</td><td>333</td><td>430</td></tr>\n",
       "\t<tr><th scope=row>LOC111128944</th><td>  0</td><td>  0</td><td>  0</td><td>  0</td><td>  0</td><td>  0</td><td>  0</td><td>  0</td><td>  0</td><td>  0</td><td>⋯</td><td>  0</td><td>  0</td><td>  0</td><td>  0</td><td>  0</td><td>  0</td><td>  0</td><td>  1</td><td>  0</td><td>  0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 119\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & B1\\_B1\\_O01 & B1\\_Nu\\_O03 & B1\\_W5\\_O50 & B2\\_B5\\_O51 & B2\\_C4\\_O40 & B2\\_Nu\\_O12 & B3\\_B4\\_O41 & B3\\_C3\\_O30 & B3\\_C6\\_O66 & B3\\_H4\\_O41 & ⋯ & W5\\_C4\\_G45 & W5\\_H4\\_G46 & W5\\_W2\\_G22 & W6\\_B3\\_G35 & W6\\_B4\\_G48 & W6\\_H6\\_G71 & W6\\_Nu\\_G41 & W6\\_Nu\\_G45 & W6\\_W3\\_G36 & W6\\_W4\\_G48\\\\\n",
       "  & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & ⋯ & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\tLOC111116054 &   2 &   1 &   6 &  10 &   9 &  16 &   0 &  10 &  22 &   9 & ⋯ &   9 &   2 &   8 &   5 &   2 &  19 &   1 &   2 &   1 &   0\\\\\n",
       "\tLOC111126949 & 885 & 652 & 477 & 654 & 586 & 523 & 392 & 357 & 496 & 540 & ⋯ & 407 & 740 & 707 & 406 & 418 & 424 & 492 & 330 & 281 & 599\\\\\n",
       "\tLOC111110729 &  64 & 209 &  93 &  63 & 100 & 177 &  76 &  98 & 235 & 181 & ⋯ &  70 & 149 & 121 & 115 & 126 & 108 & 118 & 115 & 127 & 213\\\\\n",
       "\tLOC111112434 &  11 &   7 &   2 &   0 &   2 &   2 &  12 &  15 &  10 &  36 & ⋯ &  11 &   6 &   4 &   0 &   2 &  16 &  22 &   0 &   8 &   0\\\\\n",
       "\tLOC111120752 & 360 & 586 & 336 & 426 & 351 & 417 & 236 & 278 & 761 & 373 & ⋯ & 359 & 345 & 438 & 278 & 287 & 416 & 621 & 251 & 333 & 430\\\\\n",
       "\tLOC111128944 &   0 &   0 &   0 &   0 &   0 &   0 &   0 &   0 &   0 &   0 & ⋯ &   0 &   0 &   0 &   0 &   0 &   0 &   0 &   1 &   0 &   0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 119\n",
       "\n",
       "| <!--/--> | B1_B1_O01 &lt;int&gt; | B1_Nu_O03 &lt;int&gt; | B1_W5_O50 &lt;int&gt; | B2_B5_O51 &lt;int&gt; | B2_C4_O40 &lt;int&gt; | B2_Nu_O12 &lt;int&gt; | B3_B4_O41 &lt;int&gt; | B3_C3_O30 &lt;int&gt; | B3_C6_O66 &lt;int&gt; | B3_H4_O41 &lt;int&gt; | ⋯ ⋯ | W5_C4_G45 &lt;int&gt; | W5_H4_G46 &lt;int&gt; | W5_W2_G22 &lt;int&gt; | W6_B3_G35 &lt;int&gt; | W6_B4_G48 &lt;int&gt; | W6_H6_G71 &lt;int&gt; | W6_Nu_G41 &lt;int&gt; | W6_Nu_G45 &lt;int&gt; | W6_W3_G36 &lt;int&gt; | W6_W4_G48 &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| LOC111116054 |   2 |   1 |   6 |  10 |   9 |  16 |   0 |  10 |  22 |   9 | ⋯ |   9 |   2 |   8 |   5 |   2 |  19 |   1 |   2 |   1 |   0 |\n",
       "| LOC111126949 | 885 | 652 | 477 | 654 | 586 | 523 | 392 | 357 | 496 | 540 | ⋯ | 407 | 740 | 707 | 406 | 418 | 424 | 492 | 330 | 281 | 599 |\n",
       "| LOC111110729 |  64 | 209 |  93 |  63 | 100 | 177 |  76 |  98 | 235 | 181 | ⋯ |  70 | 149 | 121 | 115 | 126 | 108 | 118 | 115 | 127 | 213 |\n",
       "| LOC111112434 |  11 |   7 |   2 |   0 |   2 |   2 |  12 |  15 |  10 |  36 | ⋯ |  11 |   6 |   4 |   0 |   2 |  16 |  22 |   0 |   8 |   0 |\n",
       "| LOC111120752 | 360 | 586 | 336 | 426 | 351 | 417 | 236 | 278 | 761 | 373 | ⋯ | 359 | 345 | 438 | 278 | 287 | 416 | 621 | 251 | 333 | 430 |\n",
       "| LOC111128944 |   0 |   0 |   0 |   0 |   0 |   0 |   0 |   0 |   0 |   0 | ⋯ |   0 |   0 |   0 |   0 |   0 |   0 |   0 |   1 |   0 |   0 |\n",
       "\n"
      ],
      "text/plain": [
       "             B1_B1_O01 B1_Nu_O03 B1_W5_O50 B2_B5_O51 B2_C4_O40 B2_Nu_O12\n",
       "LOC111116054   2         1         6        10         9        16      \n",
       "LOC111126949 885       652       477       654       586       523      \n",
       "LOC111110729  64       209        93        63       100       177      \n",
       "LOC111112434  11         7         2         0         2         2      \n",
       "LOC111120752 360       586       336       426       351       417      \n",
       "LOC111128944   0         0         0         0         0         0      \n",
       "             B3_B4_O41 B3_C3_O30 B3_C6_O66 B3_H4_O41 ⋯ W5_C4_G45 W5_H4_G46\n",
       "LOC111116054   0        10        22         9       ⋯   9         2      \n",
       "LOC111126949 392       357       496       540       ⋯ 407       740      \n",
       "LOC111110729  76        98       235       181       ⋯  70       149      \n",
       "LOC111112434  12        15        10        36       ⋯  11         6      \n",
       "LOC111120752 236       278       761       373       ⋯ 359       345      \n",
       "LOC111128944   0         0         0         0       ⋯   0         0      \n",
       "             W5_W2_G22 W6_B3_G35 W6_B4_G48 W6_H6_G71 W6_Nu_G41 W6_Nu_G45\n",
       "LOC111116054   8         5         2        19         1         2      \n",
       "LOC111126949 707       406       418       424       492       330      \n",
       "LOC111110729 121       115       126       108       118       115      \n",
       "LOC111112434   4         0         2        16        22         0      \n",
       "LOC111120752 438       278       287       416       621       251      \n",
       "LOC111128944   0         0         0         0         0         1      \n",
       "             W6_W3_G36 W6_W4_G48\n",
       "LOC111116054   1         0      \n",
       "LOC111126949 281       599      \n",
       "LOC111110729 127       213      \n",
       "LOC111112434   8         0      \n",
       "LOC111120752 333       430      \n",
       "LOC111128944   0         0      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "count.matrix <- read.csv('/work/pi_sarah_gignouxwolfsohn_uml_edu/julia_mcdonough_student_uml_edu/ce24_rnaseq/featureCounts/featureCounts_matrix.csv') %>%\n",
    "select(-Length, -B3_Nu_O24) %>%\n",
    "column_to_rownames('Gene_ID')\n",
    "\n",
    "dim(count.matrix)\n",
    "head(count.matrix)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "954f1810-47ce-4f3f-9c02-f39110637b1e",
   "metadata": {},
   "source": [
    "### meta data\n",
    "the meta data csv was already generated in [deseq_p1.v.p1](https://github.com/jgmcdonough/CE24_RNA-seq/blob/main/analysis/diff_expression/phase1_v_phase1/deseq_p1.v.p1.ipynb), so just reading in here"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "88d7ef2d-5eb0-4fb2-b256-29d93be59cf3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 9</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Phase1_treatment</th><th scope=col>Phase1_temp</th><th scope=col>Phase1_DO</th><th scope=col>Phase1_TankRep</th><th scope=col>Phase2_treatment</th><th scope=col>Phase2_temp</th><th scope=col>Phase2_DO</th><th scope=col>Phase2_TankRep</th><th scope=col>complete_trtmt</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>B1_B1_O01</th><td>both</td><td>warm</td><td>hypoxic</td><td>1</td><td>both   </td><td>warm   </td><td>hypoxic </td><td> 1</td><td>both both   </td></tr>\n",
       "\t<tr><th scope=row>B1_Nu_O03</th><td>both</td><td>warm</td><td>hypoxic</td><td>1</td><td>NA     </td><td>NA     </td><td>NA      </td><td>NA</td><td>both NA     </td></tr>\n",
       "\t<tr><th scope=row>B1_W5_O50</th><td>both</td><td>warm</td><td>hypoxic</td><td>1</td><td>warm   </td><td>warm   </td><td>normoxic</td><td> 5</td><td>both warm   </td></tr>\n",
       "\t<tr><th scope=row>B2_B5_O51</th><td>both</td><td>warm</td><td>hypoxic</td><td>2</td><td>both   </td><td>warm   </td><td>hypoxic </td><td> 5</td><td>both both   </td></tr>\n",
       "\t<tr><th scope=row>B2_C4_O40</th><td>both</td><td>warm</td><td>hypoxic</td><td>2</td><td>control</td><td>ambient</td><td>normoxic</td><td> 4</td><td>both control</td></tr>\n",
       "\t<tr><th scope=row>B2_Nu_O12</th><td>both</td><td>warm</td><td>hypoxic</td><td>2</td><td>NA     </td><td>NA     </td><td>NA      </td><td>NA</td><td>both NA     </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 9\n",
       "\\begin{tabular}{r|lllllllll}\n",
       "  & Phase1\\_treatment & Phase1\\_temp & Phase1\\_DO & Phase1\\_TankRep & Phase2\\_treatment & Phase2\\_temp & Phase2\\_DO & Phase2\\_TankRep & complete\\_trtmt\\\\\n",
       "  & <chr> & <chr> & <chr> & <int> & <chr> & <chr> & <chr> & <int> & <chr>\\\\\n",
       "\\hline\n",
       "\tB1\\_B1\\_O01 & both & warm & hypoxic & 1 & both    & warm    & hypoxic  &  1 & both both   \\\\\n",
       "\tB1\\_Nu\\_O03 & both & warm & hypoxic & 1 & NA      & NA      & NA       & NA & both NA     \\\\\n",
       "\tB1\\_W5\\_O50 & both & warm & hypoxic & 1 & warm    & warm    & normoxic &  5 & both warm   \\\\\n",
       "\tB2\\_B5\\_O51 & both & warm & hypoxic & 2 & both    & warm    & hypoxic  &  5 & both both   \\\\\n",
       "\tB2\\_C4\\_O40 & both & warm & hypoxic & 2 & control & ambient & normoxic &  4 & both control\\\\\n",
       "\tB2\\_Nu\\_O12 & both & warm & hypoxic & 2 & NA      & NA      & NA       & NA & both NA     \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 9\n",
       "\n",
       "| <!--/--> | Phase1_treatment &lt;chr&gt; | Phase1_temp &lt;chr&gt; | Phase1_DO &lt;chr&gt; | Phase1_TankRep &lt;int&gt; | Phase2_treatment &lt;chr&gt; | Phase2_temp &lt;chr&gt; | Phase2_DO &lt;chr&gt; | Phase2_TankRep &lt;int&gt; | complete_trtmt &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|\n",
       "| B1_B1_O01 | both | warm | hypoxic | 1 | both    | warm    | hypoxic  |  1 | both both    |\n",
       "| B1_Nu_O03 | both | warm | hypoxic | 1 | NA      | NA      | NA       | NA | both NA      |\n",
       "| B1_W5_O50 | both | warm | hypoxic | 1 | warm    | warm    | normoxic |  5 | both warm    |\n",
       "| B2_B5_O51 | both | warm | hypoxic | 2 | both    | warm    | hypoxic  |  5 | both both    |\n",
       "| B2_C4_O40 | both | warm | hypoxic | 2 | control | ambient | normoxic |  4 | both control |\n",
       "| B2_Nu_O12 | both | warm | hypoxic | 2 | NA      | NA      | NA       | NA | both NA      |\n",
       "\n"
      ],
      "text/plain": [
       "          Phase1_treatment Phase1_temp Phase1_DO Phase1_TankRep\n",
       "B1_B1_O01 both             warm        hypoxic   1             \n",
       "B1_Nu_O03 both             warm        hypoxic   1             \n",
       "B1_W5_O50 both             warm        hypoxic   1             \n",
       "B2_B5_O51 both             warm        hypoxic   2             \n",
       "B2_C4_O40 both             warm        hypoxic   2             \n",
       "B2_Nu_O12 both             warm        hypoxic   2             \n",
       "          Phase2_treatment Phase2_temp Phase2_DO Phase2_TankRep complete_trtmt\n",
       "B1_B1_O01 both             warm        hypoxic    1             both both     \n",
       "B1_Nu_O03 NA               NA          NA        NA             both NA       \n",
       "B1_W5_O50 warm             warm        normoxic   5             both warm     \n",
       "B2_B5_O51 both             warm        hypoxic    5             both both     \n",
       "B2_C4_O40 control          ambient     normoxic   4             both control  \n",
       "B2_Nu_O12 NA               NA          NA        NA             both NA       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "meta <- read.csv('/project/pi_sarah_gignouxwolfsohn_uml_edu/julia/CE_2024/CE24_RNA-seq/sample_metaData.csv') %>%\n",
    "column_to_rownames('Sample') %>%\n",
    "mutate(complete_trtmt = paste(Phase1_treatment, Phase2_treatment))\n",
    "head(meta)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "7d8c2ae1-7528-447a-8ac5-af9f6b9d6bb2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "TRUE"
      ],
      "text/latex": [
       "TRUE"
      ],
      "text/markdown": [
       "TRUE"
      ],
      "text/plain": [
       "[1] TRUE"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# All samples\n",
    "meta <- meta[colnames(count.matrix),]\n",
    "all(rownames(meta) == colnames(count.matrix))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "56a2ebf7-eedb-4214-b1a8-3b96e4b14714",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "TRUE"
      ],
      "text/latex": [
       "TRUE"
      ],
      "text/markdown": [
       "TRUE"
      ],
      "text/plain": [
       "[1] TRUE"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Phase 1 samples\n",
    "meta.p1 <- meta %>%\n",
    "filter(is.na(Phase2_treatment))\n",
    "\n",
    "counts.p1 <- count.matrix[,rownames(meta.p1)]\n",
    "all(rownames(meta.p1) == colnames(counts.p1))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "75764cb3-17a9-4785-85ab-afeeddbf9eec",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "TRUE"
      ],
      "text/latex": [
       "TRUE"
      ],
      "text/markdown": [
       "TRUE"
      ],
      "text/plain": [
       "[1] TRUE"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Phase 2 samples\n",
    "meta.p2 <- meta %>%\n",
    "filter(!is.na(Phase2_treatment))\n",
    "\n",
    "counts.p2 <- count.matrix[,rownames(meta.p2)]\n",
    "all(rownames(meta.p2) == colnames(counts.p2))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7e0aab14-364c-4dbc-a43b-6902f59c5718",
   "metadata": {},
   "source": [
    "following [this code notebook](https://github.com/SamGurr/Pgenerosa_OA_TagSeq/blob/main/TagSeq/Analysis/Scripts/Day0_WGCNA_all.R) from Gurr et al., 2022\n",
    "\n",
    "## 2. Phase 1 samples only\n",
    "\n",
    "### A. create DESeq object"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "6b074d88-c918-47f4-86ff-3850ca77963f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>23</li><li>38838</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 23\n",
       "\\item 38838\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 23\n",
       "2. 38838\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1]    23 38838"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dds.p1 <- DESeqDataSetFromMatrix(countData = counts.p1,\n",
    "                              colData = meta.p1,\n",
    "                              design = ~ 1) # don't need design, just need an object to transform\n",
    "\n",
    "# transoform data\n",
    "dds.p1_vst <- vst(dds.p1)\n",
    "dds.p1_vst <- assay(dds.p1_vst) # call only the transformed coutns in the dds object\n",
    "# fix(dds.d0_vst)\n",
    "dds.p1_vst <- t(dds.p1_vst) # transpose columns to rows and vice versa\n",
    "\n",
    "dim(dds.p1_vst)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "c11a023e-3595-43df-9974-7836294efae7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " Flagging genes and samples with too many missing values...\n",
      "  ..step 1\n",
      "  ..Excluding 3857 genes from the calculation due to too many missing samples or zero variance.\n",
      "  ..step 2\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "FALSE"
      ],
      "text/latex": [
       "FALSE"
      ],
      "text/markdown": [
       "FALSE"
      ],
      "text/plain": [
       "[1] FALSE"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "gsg = goodSamplesGenes(dds.p1_vst, verbose = 3)\n",
    "gsg$allOK"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R [conda env:.conda-r-env]",
   "language": "R",
   "name": "conda-env-.conda-r-env-r"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.3.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
