{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "534c3a84-d662-4850-991d-1caff76528b5",
   "metadata": {},
   "source": [
    "# Gene Set Enrichment Analysis (GSEA)\n",
    "# Phase 1 vs. Phase 1\n",
    "\n",
    "using [`GSEA`](https://rdrr.io/bioc/clusterProfiler/man/GSEA.html) and [`enrichr`](https://rdrr.io/bioc/clusterProfiler/man/enricher.html) commands in the [clusterProfiler](https://github.com/YuLab-SMU/clusterProfiler) R package"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c1bb115a-2333-4537-a40d-85d72eaff270",
   "metadata": {},
   "source": [
    "## 0. load libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "068d6f78-2e76-492f-8dec-2f8931380c70",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DOSE v3.28.2  For help: https://yulab-smu.top/biomedical-knowledge-mining-book/\n",
      "\n",
      "If you use DOSE in published research, please cite:\n",
      "Guangchuang Yu, Li-Gen Wang, Guang-Rong Yan, Qing-Yu He. DOSE: an R/Bioconductor package for Disease Ontology Semantic and Enrichment analysis. Bioinformatics 2015, 31(4):608-609\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(tidyverse)\n",
    "library(clusterProfiler)\n",
    "library(enrichplot)\n",
    "library(DOSE)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ecce72cc-aaf7-4ea4-bf96-b482f2888d68",
   "metadata": {},
   "source": [
    "## 1. read in CSVs"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "45b9660b-739a-43ce-af64-12073e56d0be",
   "metadata": {},
   "source": [
    "read in GO annotation data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "713e8f0d-826c-4fde-a948-a2d0079d70c3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>gene</th><th scope=col>Gene.Ontology.IDs</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>LOC111126949</td><td>GO:0005042; GO:0005737; GO:0008233; GO:0016020; GO:0043123; GO:0046330                                                </td></tr>\n",
       "\t<tr><th scope=row>2</th><td>LOC111112434</td><td>GO:0000981; GO:0003700; GO:0005634; GO:0016607; GO:0043565; GO:0045944                                                </td></tr>\n",
       "\t<tr><th scope=row>3</th><td>LOC111120752</td><td>GO:0004750; GO:0005829; GO:0005975; GO:0006098; GO:0009052; GO:0042802; GO:0042803; GO:0046872; GO:0070062            </td></tr>\n",
       "\t<tr><th scope=row>4</th><td>LOC111105685</td><td>GO:0003682; GO:0004518; GO:0005634; GO:0035098; GO:0035102; GO:0040029; GO:0046872                                    </td></tr>\n",
       "\t<tr><th scope=row>5</th><td>LOC111113860</td><td>GO:0004062; GO:0005737; GO:0006068; GO:0006805; GO:0008146; GO:0009812; GO:0030855; GO:0042403; GO:0050427; GO:0051923</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>LOC111109550</td><td>GO:0004062; GO:0005737; GO:0005764; GO:0008146; GO:0051923                                                            </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 2\n",
       "\\begin{tabular}{r|ll}\n",
       "  & gene & Gene.Ontology.IDs\\\\\n",
       "  & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & LOC111126949 & GO:0005042; GO:0005737; GO:0008233; GO:0016020; GO:0043123; GO:0046330                                                \\\\\n",
       "\t2 & LOC111112434 & GO:0000981; GO:0003700; GO:0005634; GO:0016607; GO:0043565; GO:0045944                                                \\\\\n",
       "\t3 & LOC111120752 & GO:0004750; GO:0005829; GO:0005975; GO:0006098; GO:0009052; GO:0042802; GO:0042803; GO:0046872; GO:0070062            \\\\\n",
       "\t4 & LOC111105685 & GO:0003682; GO:0004518; GO:0005634; GO:0035098; GO:0035102; GO:0040029; GO:0046872                                    \\\\\n",
       "\t5 & LOC111113860 & GO:0004062; GO:0005737; GO:0006068; GO:0006805; GO:0008146; GO:0009812; GO:0030855; GO:0042403; GO:0050427; GO:0051923\\\\\n",
       "\t6 & LOC111109550 & GO:0004062; GO:0005737; GO:0005764; GO:0008146; GO:0051923                                                            \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 2\n",
       "\n",
       "| <!--/--> | gene &lt;chr&gt; | Gene.Ontology.IDs &lt;chr&gt; |\n",
       "|---|---|---|\n",
       "| 1 | LOC111126949 | GO:0005042; GO:0005737; GO:0008233; GO:0016020; GO:0043123; GO:0046330                                                 |\n",
       "| 2 | LOC111112434 | GO:0000981; GO:0003700; GO:0005634; GO:0016607; GO:0043565; GO:0045944                                                 |\n",
       "| 3 | LOC111120752 | GO:0004750; GO:0005829; GO:0005975; GO:0006098; GO:0009052; GO:0042802; GO:0042803; GO:0046872; GO:0070062             |\n",
       "| 4 | LOC111105685 | GO:0003682; GO:0004518; GO:0005634; GO:0035098; GO:0035102; GO:0040029; GO:0046872                                     |\n",
       "| 5 | LOC111113860 | GO:0004062; GO:0005737; GO:0006068; GO:0006805; GO:0008146; GO:0009812; GO:0030855; GO:0042403; GO:0050427; GO:0051923 |\n",
       "| 6 | LOC111109550 | GO:0004062; GO:0005737; GO:0005764; GO:0008146; GO:0051923                                                             |\n",
       "\n"
      ],
      "text/plain": [
       "  gene        \n",
       "1 LOC111126949\n",
       "2 LOC111112434\n",
       "3 LOC111120752\n",
       "4 LOC111105685\n",
       "5 LOC111113860\n",
       "6 LOC111109550\n",
       "  Gene.Ontology.IDs                                                                                                     \n",
       "1 GO:0005042; GO:0005737; GO:0008233; GO:0016020; GO:0043123; GO:0046330                                                \n",
       "2 GO:0000981; GO:0003700; GO:0005634; GO:0016607; GO:0043565; GO:0045944                                                \n",
       "3 GO:0004750; GO:0005829; GO:0005975; GO:0006098; GO:0009052; GO:0042802; GO:0042803; GO:0046872; GO:0070062            \n",
       "4 GO:0003682; GO:0004518; GO:0005634; GO:0035098; GO:0035102; GO:0040029; GO:0046872                                    \n",
       "5 GO:0004062; GO:0005737; GO:0006068; GO:0006805; GO:0008146; GO:0009812; GO:0030855; GO:0042403; GO:0050427; GO:0051923\n",
       "6 GO:0004062; GO:0005737; GO:0005764; GO:0008146; GO:0051923                                                            "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# col1 = gene ID\n",
    "# col2 = GO ID \n",
    "gene2go <- read.csv('/work/pi_sarah_gignouxwolfsohn_uml_edu/julia_mcdonough_student_uml_edu/ref_files/annotations/geneGO.txt', sep = '\\t')\n",
    "head(gene2go)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8b92bf06-6dca-4236-9865-a53624e6c7be",
   "metadata": {},
   "source": [
    "re-format for correct input for `GSEA()` - two columns, one for GO term and one for gene ID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "cd39e6eb-8ac0-485e-9c51-b86e3f1d6b91",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'tbl_df'</li><li>'tbl'</li><li>'data.frame'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'tbl\\_df'\n",
       "\\item 'tbl'\n",
       "\\item 'data.frame'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'tbl_df'\n",
       "2. 'tbl'\n",
       "3. 'data.frame'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"tbl_df\"     \"tbl\"        \"data.frame\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "tibble [223,103 × 2] (S3: tbl_df/tbl/data.frame)\n",
      " $ term: chr [1:223103] \"GO:0005042\" \"GO:0005737\" \"GO:0008233\" \"GO:0016020\" ...\n",
      " $ gene: chr [1:223103] \"LOC111126949\" \"LOC111126949\" \"LOC111126949\" \"LOC111126949\" ...\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A tibble: 6 × 2</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>term</th><th scope=col>gene</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>GO:0005042</td><td>LOC111126949</td></tr>\n",
       "\t<tr><td>GO:0005737</td><td>LOC111126949</td></tr>\n",
       "\t<tr><td>GO:0008233</td><td>LOC111126949</td></tr>\n",
       "\t<tr><td>GO:0016020</td><td>LOC111126949</td></tr>\n",
       "\t<tr><td>GO:0043123</td><td>LOC111126949</td></tr>\n",
       "\t<tr><td>GO:0046330</td><td>LOC111126949</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A tibble: 6 × 2\n",
       "\\begin{tabular}{ll}\n",
       " term & gene\\\\\n",
       " <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t GO:0005042 & LOC111126949\\\\\n",
       "\t GO:0005737 & LOC111126949\\\\\n",
       "\t GO:0008233 & LOC111126949\\\\\n",
       "\t GO:0016020 & LOC111126949\\\\\n",
       "\t GO:0043123 & LOC111126949\\\\\n",
       "\t GO:0046330 & LOC111126949\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A tibble: 6 × 2\n",
       "\n",
       "| term &lt;chr&gt; | gene &lt;chr&gt; |\n",
       "|---|---|\n",
       "| GO:0005042 | LOC111126949 |\n",
       "| GO:0005737 | LOC111126949 |\n",
       "| GO:0008233 | LOC111126949 |\n",
       "| GO:0016020 | LOC111126949 |\n",
       "| GO:0043123 | LOC111126949 |\n",
       "| GO:0046330 | LOC111126949 |\n",
       "\n"
      ],
      "text/plain": [
       "  term       gene        \n",
       "1 GO:0005042 LOC111126949\n",
       "2 GO:0005737 LOC111126949\n",
       "3 GO:0008233 LOC111126949\n",
       "4 GO:0016020 LOC111126949\n",
       "5 GO:0043123 LOC111126949\n",
       "6 GO:0046330 LOC111126949"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "term2gene <- gene2go %>%\n",
    "  mutate(GO_terms = strsplit(Gene.Ontology.IDs, \",\\\\s*|;\\\\s*|`\")) %>%  # Split by comma, semicolon, or backtick\n",
    "  unnest(GO_terms) %>%\n",
    "  filter(grepl(\"^GO:\", GO_terms)) %>%  # Keep only valid GO terms\n",
    "  select(term = GO_terms, gene = gene)\n",
    "\n",
    "\n",
    "\n",
    "class(term2gene)\n",
    "str(term2gene)\n",
    "head(term2gene)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "71637029-d6f6-4825-8cf5-2056455de438",
   "metadata": {},
   "source": [
    "read in *all* genes from DESeq output (not just DEGs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "cdccdd59-ec00-44e2-be72-b2fb40099dda",
   "metadata": {},
   "outputs": [],
   "source": [
    "##### comparisons vs. control\n",
    "# warm vs. control\n",
    "w.c <- read.csv('/project/pi_sarah_gignouxwolfsohn_uml_edu/julia/CE_2024/CE24_RNA-seq/analysis/diff_expression/phase1_v_phase1/deseq_res_files/p1.warm_v_control.csv')\n",
    "\n",
    "# both vs. control\n",
    "b.c <- read.csv('/project/pi_sarah_gignouxwolfsohn_uml_edu/julia/CE_2024/CE24_RNA-seq/analysis/diff_expression/phase1_v_phase1/deseq_res_files/p1.both_v_control.csv')\n",
    "\n",
    "# hyp vs. control\n",
    "h.c <- read.csv('/project/pi_sarah_gignouxwolfsohn_uml_edu/julia/CE_2024/CE24_RNA-seq/analysis/diff_expression/phase1_v_phase1/deseq_res_files/p1.hyp_v_control.csv')\n",
    "\n",
    "##### other comparisons\n",
    "# hyp vs. both\n",
    "h.b <- read.csv('/project/pi_sarah_gignouxwolfsohn_uml_edu/julia/CE_2024/CE24_RNA-seq/analysis/diff_expression/phase1_v_phase1/deseq_res_files/p1.hyp_v_both.csv')\n",
    "\n",
    "# warm vs. both\n",
    "w.b <- read.csv('/project/pi_sarah_gignouxwolfsohn_uml_edu/julia/CE_2024/CE24_RNA-seq/analysis/diff_expression/phase1_v_phase1/deseq_res_files/p1.warm_v_both.csv')\n",
    "\n",
    "# warm vs. hyp\n",
    "w.h <- read.csv('/project/pi_sarah_gignouxwolfsohn_uml_edu/julia/CE_2024/CE24_RNA-seq/analysis/diff_expression/phase1_v_phase1/deseq_res_files/p1.warm_v_hyp.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b04de9cc-65a6-4a17-add1-86a3b1a09d20",
   "metadata": {},
   "source": [
    "## 2. Formatting CSVs\n",
    "input to `gsea()` is an order ranked geneList - I'm going to rank by log2FoldChange"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "a2f4f18e-3563-45a1-8250-40053d288f7a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# data needs to be a numeric vector\n",
    "# create function for above cell block so can easily apply to other data\n",
    "format_geneList <- function(df) {\n",
    "    geneList <- df$log2FoldChange \n",
    "    # Set gene IDs as names\n",
    "    names(geneList) <- df$X\n",
    "    \n",
    "    # Sort in decreasing order\n",
    "    geneList <- sort(geneList, decreasing = TRUE)\n",
    "\n",
    "    return(geneList)\n",
    "    }"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "926afae0-eb39-4fa8-a764-2640b1abebbe",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "'numeric'"
      ],
      "text/latex": [
       "'numeric'"
      ],
      "text/markdown": [
       "'numeric'"
      ],
      "text/plain": [
       "[1] \"numeric\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".dl-inline {width: auto; margin:0; padding: 0}\n",
       ".dl-inline>dt, .dl-inline>dd {float: none; width: auto; display: inline-block}\n",
       ".dl-inline>dt::after {content: \":\\0020\"; padding-right: .5ex}\n",
       ".dl-inline>dt:not(:first-of-type) {padding-left: .5ex}\n",
       "</style><dl class=dl-inline><dt>LOC111115187</dt><dd>25.3711708463482</dd><dt>LOC111137749</dt><dd>23.5894690567884</dd><dt>LOC111105825</dt><dd>23.4048299729129</dd><dt>LOC111117559</dt><dd>22.4890967388319</dd><dt>LOC111105743</dt><dd>22.4176052350207</dd><dt>LOC111104227</dt><dd>22.1552368480927</dd></dl>\n"
      ],
      "text/latex": [
       "\\begin{description*}\n",
       "\\item[LOC111115187] 25.3711708463482\n",
       "\\item[LOC111137749] 23.5894690567884\n",
       "\\item[LOC111105825] 23.4048299729129\n",
       "\\item[LOC111117559] 22.4890967388319\n",
       "\\item[LOC111105743] 22.4176052350207\n",
       "\\item[LOC111104227] 22.1552368480927\n",
       "\\end{description*}\n"
      ],
      "text/markdown": [
       "LOC111115187\n",
       ":   25.3711708463482LOC111137749\n",
       ":   23.5894690567884LOC111105825\n",
       ":   23.4048299729129LOC111117559\n",
       ":   22.4890967388319LOC111105743\n",
       ":   22.4176052350207LOC111104227\n",
       ":   22.1552368480927\n",
       "\n"
      ],
      "text/plain": [
       "LOC111115187 LOC111137749 LOC111105825 LOC111117559 LOC111105743 LOC111104227 \n",
       "    25.37117     23.58947     23.40483     22.48910     22.41761     22.15524 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "w.c_geneList <- format_geneList(w.c)\n",
    "h.c_geneList <- format_geneList(h.c)\n",
    "b.c_geneList <- format_geneList(b.c)\n",
    "h.b_geneList <- format_geneList(h.b)\n",
    "w.b_geneList <- format_geneList(w.b)\n",
    "w.h_geneList <- format_geneList(w.h)\n",
    "\n",
    "# check that it worked\n",
    "class(w.c_geneList)\n",
    "head(w.c_geneList)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f44fba48-4752-4dfd-99f5-8fd914225195",
   "metadata": {},
   "source": [
    "## 3. Run GSEA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "e4f3276c-252c-46c0-be6e-b697bd704ea4",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "preparing geneSet collections...\n",
      "\n",
      "GSEA analysis...\n",
      "\n",
      "Warning message in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, :\n",
      "“There are ties in the preranked stats (1.24% of the list).\n",
      "The order of those tied genes will be arbitrary, which may produce unexpected results.”\n",
      "no term enriched under specific pvalueCutoff...\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 0 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>ID</th><th scope=col>Description</th><th scope=col>setSize</th><th scope=col>enrichmentScore</th><th scope=col>NES</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 0 × 8\n",
       "\\begin{tabular}{llllllll}\n",
       " ID & Description & setSize & enrichmentScore & NES & pvalue & p.adjust & qvalue\\\\\n",
       " <chr> & <chr> & <int> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 0 × 8\n",
       "\n",
       "| ID &lt;chr&gt; | Description &lt;chr&gt; | setSize &lt;int&gt; | enrichmentScore &lt;dbl&gt; | NES &lt;dbl&gt; | pvalue &lt;dbl&gt; | p.adjust &lt;dbl&gt; | qvalue &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|\n",
       "\n"
      ],
      "text/plain": [
       "     ID Description setSize enrichmentScore NES pvalue p.adjust qvalue"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# warm vs control\n",
    "gsea_res_w.c <- GSEA(\n",
    "    geneList = w.c_geneList,\n",
    "    TERM2GENE = term2gene, \n",
    "    pvalueCutoff = 0.05)\n",
    "as.data.frame(gsea_res_w.c)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "6c44e6ce-884c-4736-9450-4c5a83a117ce",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "preparing geneSet collections...\n",
      "\n",
      "GSEA analysis...\n",
      "\n",
      "Warning message in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, :\n",
      "“There are ties in the preranked stats (1.34% of the list).\n",
      "The order of those tied genes will be arbitrary, which may produce unexpected results.”\n",
      "leading edge analysis...\n",
      "\n",
      "done...\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 1 × 11</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>setSize</th><th scope=col>enrichmentScore</th><th scope=col>NES</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>rank</th><th scope=col>leading_edge</th><th scope=col>core_enrichment</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>GO:0050770</th><td>GO:0050770</td><td>GO:0050770</td><td>19</td><td>-0.9982205</td><td>-1.441147</td><td>5.635311e-06</td><td>0.02061397</td><td>0.02061397</td><td>22</td><td>tags=16%, list=0%, signal=16%</td><td>LOC111130881/LOC111125415/LOC111125416</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 1 × 11\n",
       "\\begin{tabular}{r|lllllllllll}\n",
       "  & ID & Description & setSize & enrichmentScore & NES & pvalue & p.adjust & qvalue & rank & leading\\_edge & core\\_enrichment\\\\\n",
       "  & <chr> & <chr> & <int> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\tGO:0050770 & GO:0050770 & GO:0050770 & 19 & -0.9982205 & -1.441147 & 5.635311e-06 & 0.02061397 & 0.02061397 & 22 & tags=16\\%, list=0\\%, signal=16\\% & LOC111130881/LOC111125415/LOC111125416\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 1 × 11\n",
       "\n",
       "| <!--/--> | ID &lt;chr&gt; | Description &lt;chr&gt; | setSize &lt;int&gt; | enrichmentScore &lt;dbl&gt; | NES &lt;dbl&gt; | pvalue &lt;dbl&gt; | p.adjust &lt;dbl&gt; | qvalue &lt;dbl&gt; | rank &lt;dbl&gt; | leading_edge &lt;chr&gt; | core_enrichment &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| GO:0050770 | GO:0050770 | GO:0050770 | 19 | -0.9982205 | -1.441147 | 5.635311e-06 | 0.02061397 | 0.02061397 | 22 | tags=16%, list=0%, signal=16% | LOC111130881/LOC111125415/LOC111125416 |\n",
       "\n"
      ],
      "text/plain": [
       "           ID         Description setSize enrichmentScore NES      \n",
       "GO:0050770 GO:0050770 GO:0050770  19      -0.9982205      -1.441147\n",
       "           pvalue       p.adjust   qvalue     rank\n",
       "GO:0050770 5.635311e-06 0.02061397 0.02061397 22  \n",
       "           leading_edge                  core_enrichment                       \n",
       "GO:0050770 tags=16%, list=0%, signal=16% LOC111130881/LOC111125415/LOC111125416"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# both vs control\n",
    "gsea_res_b.c <- GSEA(\n",
    "    geneList = b.c_geneList,\n",
    "    TERM2GENE = term2gene, \n",
    "    pvalueCutoff = 0.05,\n",
    "    verbose = TRUE)\n",
    "as.data.frame(gsea_res_b.c)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "92d7170a-9948-4cea-bf3d-df7f0ea55f96",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "preparing geneSet collections...\n",
      "\n",
      "GSEA analysis...\n",
      "\n",
      "Warning message in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, :\n",
      "“There are ties in the preranked stats (1.26% of the list).\n",
      "The order of those tied genes will be arbitrary, which may produce unexpected results.”\n",
      "no term enriched under specific pvalueCutoff...\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 0 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>ID</th><th scope=col>Description</th><th scope=col>setSize</th><th scope=col>enrichmentScore</th><th scope=col>NES</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 0 × 8\n",
       "\\begin{tabular}{llllllll}\n",
       " ID & Description & setSize & enrichmentScore & NES & pvalue & p.adjust & qvalue\\\\\n",
       " <chr> & <chr> & <int> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 0 × 8\n",
       "\n",
       "| ID &lt;chr&gt; | Description &lt;chr&gt; | setSize &lt;int&gt; | enrichmentScore &lt;dbl&gt; | NES &lt;dbl&gt; | pvalue &lt;dbl&gt; | p.adjust &lt;dbl&gt; | qvalue &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|\n",
       "\n"
      ],
      "text/plain": [
       "     ID Description setSize enrichmentScore NES pvalue p.adjust qvalue"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# hypoxic vs control\n",
    "gsea_res_h.c <- GSEA(\n",
    "    geneList = h.c_geneList,\n",
    "    TERM2GENE = term2gene, \n",
    "    pvalueCutoff = 0.05,\n",
    "    verbose = TRUE)\n",
    "as.data.frame(gsea_res_h.c)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "id": "9f0e02a9-63fd-4b37-ad9d-666f53f07021",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "preparing geneSet collections...\n",
      "\n",
      "GSEA analysis...\n",
      "\n",
      "Warning message in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, :\n",
      "“There are ties in the preranked stats (0.77% of the list).\n",
      "The order of those tied genes will be arbitrary, which may produce unexpected results.”\n",
      "leading edge analysis...\n",
      "\n",
      "done...\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 1 × 11</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>setSize</th><th scope=col>enrichmentScore</th><th scope=col>NES</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>rank</th><th scope=col>leading_edge</th><th scope=col>core_enrichment</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>GO:0007274</th><td>GO:0007274</td><td>GO:0007274</td><td>29</td><td>0.9989389</td><td>1.396448</td><td>1.240006e-05</td><td>0.04535941</td><td>0.04535941</td><td>10</td><td>tags=3%, list=0%, signal=3%</td><td>LOC111120179</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 1 × 11\n",
       "\\begin{tabular}{r|lllllllllll}\n",
       "  & ID & Description & setSize & enrichmentScore & NES & pvalue & p.adjust & qvalue & rank & leading\\_edge & core\\_enrichment\\\\\n",
       "  & <chr> & <chr> & <int> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <int> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\tGO:0007274 & GO:0007274 & GO:0007274 & 29 & 0.9989389 & 1.396448 & 1.240006e-05 & 0.04535941 & 0.04535941 & 10 & tags=3\\%, list=0\\%, signal=3\\% & LOC111120179\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 1 × 11\n",
       "\n",
       "| <!--/--> | ID &lt;chr&gt; | Description &lt;chr&gt; | setSize &lt;int&gt; | enrichmentScore &lt;dbl&gt; | NES &lt;dbl&gt; | pvalue &lt;dbl&gt; | p.adjust &lt;dbl&gt; | qvalue &lt;dbl&gt; | rank &lt;int&gt; | leading_edge &lt;chr&gt; | core_enrichment &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| GO:0007274 | GO:0007274 | GO:0007274 | 29 | 0.9989389 | 1.396448 | 1.240006e-05 | 0.04535941 | 0.04535941 | 10 | tags=3%, list=0%, signal=3% | LOC111120179 |\n",
       "\n"
      ],
      "text/plain": [
       "           ID         Description setSize enrichmentScore NES      pvalue      \n",
       "GO:0007274 GO:0007274 GO:0007274  29      0.9989389       1.396448 1.240006e-05\n",
       "           p.adjust   qvalue     rank leading_edge               \n",
       "GO:0007274 0.04535941 0.04535941 10   tags=3%, list=0%, signal=3%\n",
       "           core_enrichment\n",
       "GO:0007274 LOC111120179   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# hypoxic vs both\n",
    "gsea_res_h.b <- GSEA(\n",
    "    geneList = h.b_geneList,\n",
    "    TERM2GENE = term2gene, \n",
    "    pvalueCutoff = 0.05,\n",
    "    verbose = TRUE)\n",
    "as.data.frame(gsea_res_h.b)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "id": "bb055d47-62f9-44e3-80e9-acca29e67cb8",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "preparing geneSet collections...\n",
      "\n",
      "GSEA analysis...\n",
      "\n",
      "Warning message in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, :\n",
      "“There are ties in the preranked stats (0.72% of the list).\n",
      "The order of those tied genes will be arbitrary, which may produce unexpected results.”\n",
      "no term enriched under specific pvalueCutoff...\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 0 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>ID</th><th scope=col>Description</th><th scope=col>setSize</th><th scope=col>enrichmentScore</th><th scope=col>NES</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 0 × 8\n",
       "\\begin{tabular}{llllllll}\n",
       " ID & Description & setSize & enrichmentScore & NES & pvalue & p.adjust & qvalue\\\\\n",
       " <chr> & <chr> & <int> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 0 × 8\n",
       "\n",
       "| ID &lt;chr&gt; | Description &lt;chr&gt; | setSize &lt;int&gt; | enrichmentScore &lt;dbl&gt; | NES &lt;dbl&gt; | pvalue &lt;dbl&gt; | p.adjust &lt;dbl&gt; | qvalue &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|---|\n",
       "\n"
      ],
      "text/plain": [
       "     ID Description setSize enrichmentScore NES pvalue p.adjust qvalue"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# warm vs both\n",
    "gsea_res_w.b <- GSEA(\n",
    "    geneList = w.b_geneList,\n",
    "    TERM2GENE = term2gene, \n",
    "    pvalueCutoff = 0.05,\n",
    "    verbose = TRUE)\n",
    "as.data.frame(gsea_res_w.b)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "28c14b39-ee8b-4b33-b44a-9269e4d86712",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "preparing geneSet collections...\n",
      "\n",
      "GSEA analysis...\n",
      "\n",
      "Warning message in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, :\n",
      "“There are ties in the preranked stats (1.48% of the list).\n",
      "The order of those tied genes will be arbitrary, which may produce unexpected results.”\n",
      "leading edge analysis...\n",
      "\n",
      "done...\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 11</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>setSize</th><th scope=col>enrichmentScore</th><th scope=col>NES</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>rank</th><th scope=col>leading_edge</th><th scope=col>core_enrichment</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>GO:0004748</th><td>GO:0004748</td><td>GO:0004748</td><td>75</td><td>0.9643563</td><td>2.184434</td><td>1.228681e-05</td><td>0.02247258</td><td>0.02247258</td><td>33</td><td>tags=1%, list=0%, signal=1%</td><td>LOC111108941</td></tr>\n",
       "\t<tr><th scope=row>GO:0009263</th><td>GO:0009263</td><td>GO:0009263</td><td>75</td><td>0.9643563</td><td>2.184434</td><td>1.228681e-05</td><td>0.02247258</td><td>0.02247258</td><td>33</td><td>tags=1%, list=0%, signal=1%</td><td>LOC111108941</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 11\n",
       "\\begin{tabular}{r|lllllllllll}\n",
       "  & ID & Description & setSize & enrichmentScore & NES & pvalue & p.adjust & qvalue & rank & leading\\_edge & core\\_enrichment\\\\\n",
       "  & <chr> & <chr> & <int> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <int> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\tGO:0004748 & GO:0004748 & GO:0004748 & 75 & 0.9643563 & 2.184434 & 1.228681e-05 & 0.02247258 & 0.02247258 & 33 & tags=1\\%, list=0\\%, signal=1\\% & LOC111108941\\\\\n",
       "\tGO:0009263 & GO:0009263 & GO:0009263 & 75 & 0.9643563 & 2.184434 & 1.228681e-05 & 0.02247258 & 0.02247258 & 33 & tags=1\\%, list=0\\%, signal=1\\% & LOC111108941\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 11\n",
       "\n",
       "| <!--/--> | ID &lt;chr&gt; | Description &lt;chr&gt; | setSize &lt;int&gt; | enrichmentScore &lt;dbl&gt; | NES &lt;dbl&gt; | pvalue &lt;dbl&gt; | p.adjust &lt;dbl&gt; | qvalue &lt;dbl&gt; | rank &lt;int&gt; | leading_edge &lt;chr&gt; | core_enrichment &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| GO:0004748 | GO:0004748 | GO:0004748 | 75 | 0.9643563 | 2.184434 | 1.228681e-05 | 0.02247258 | 0.02247258 | 33 | tags=1%, list=0%, signal=1% | LOC111108941 |\n",
       "| GO:0009263 | GO:0009263 | GO:0009263 | 75 | 0.9643563 | 2.184434 | 1.228681e-05 | 0.02247258 | 0.02247258 | 33 | tags=1%, list=0%, signal=1% | LOC111108941 |\n",
       "\n"
      ],
      "text/plain": [
       "           ID         Description setSize enrichmentScore NES      pvalue      \n",
       "GO:0004748 GO:0004748 GO:0004748  75      0.9643563       2.184434 1.228681e-05\n",
       "GO:0009263 GO:0009263 GO:0009263  75      0.9643563       2.184434 1.228681e-05\n",
       "           p.adjust   qvalue     rank leading_edge               \n",
       "GO:0004748 0.02247258 0.02247258 33   tags=1%, list=0%, signal=1%\n",
       "GO:0009263 0.02247258 0.02247258 33   tags=1%, list=0%, signal=1%\n",
       "           core_enrichment\n",
       "GO:0004748 LOC111108941   \n",
       "GO:0009263 LOC111108941   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# warm vs hypoxic\n",
    "gsea_res_w.h <- GSEA(\n",
    "    geneList = w.h_geneList,\n",
    "    TERM2GENE = term2gene, \n",
    "    pvalueCutoff = 0.05,\n",
    "    verbose = TRUE)\n",
    "as.data.frame(gsea_res_w.h)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R [conda env:.conda-r-env]",
   "language": "R",
   "name": "conda-env-.conda-r-env-r"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.3.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
