{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "f33640bb-ba14-4584-965b-393ef36eac0a",
   "metadata": {},
   "source": [
    "# KEGG Analysis\n",
    "\n",
    "# Phase 1 vs. Phase 1\n",
    "\n",
    "Using `enrichKEGG` in the clusterProfiler package to investigate enrichment of KEGG categories in our DEGs\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "332de215-bd3b-4da2-a79d-74f655f68a96",
   "metadata": {},
   "source": [
    "## 0. load libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "aa0c6a49-e72d-401f-b571-97c7ec9f9c0c",
   "metadata": {},
   "outputs": [],
   "source": [
    "library(stringr)\n",
    "library(dplyr)\n",
    "library(ggplot2)\n",
    "library(clusterProfiler)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bc1673cc-b75c-4654-9d62-84451da46f87",
   "metadata": {},
   "source": [
    "## 1. read CSVs\n",
    "\n",
    "read in as data frames"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "4070195f-5873-4f4a-b01e-84990eb1708b",
   "metadata": {},
   "outputs": [],
   "source": [
    "##### comparisons vs. control\n",
    "# warm vs. control\n",
    "w.c_deg <- read.csv('/project/pi_sarah_gignouxwolfsohn_uml_edu/julia/CE_2024/CE24_RNA-seq/analysis/diff_expression/phase1_v_phase1/deseq_res_files/DEGs/DEG_w_c.csv')\n",
    "\n",
    "# both vs. control\n",
    "b.c_deg <- read.csv('/project/pi_sarah_gignouxwolfsohn_uml_edu/julia/CE_2024/CE24_RNA-seq/analysis/diff_expression/phase1_v_phase1/deseq_res_files/DEGs/DEG_b_c.csv')\n",
    "\n",
    "# hyp vs. control\n",
    "h.c_deg <- read.csv('/project/pi_sarah_gignouxwolfsohn_uml_edu/julia/CE_2024/CE24_RNA-seq/analysis/diff_expression/phase1_v_phase1/deseq_res_files/DEGs/DEG_h_c.csv')\n",
    "\n",
    "##### other comparisons\n",
    "# hyp vs. both\n",
    "h.b_deg <- read.csv('/project/pi_sarah_gignouxwolfsohn_uml_edu/julia/CE_2024/CE24_RNA-seq/analysis/diff_expression/phase1_v_phase1/deseq_res_files/DEGs/DEG_h_b.csv')\n",
    "\n",
    "# warm vs. both\n",
    "w.b_deg <- read.csv('/project/pi_sarah_gignouxwolfsohn_uml_edu/julia/CE_2024/CE24_RNA-seq/analysis/diff_expression/phase1_v_phase1/deseq_res_files/DEGs/DEG_w_b.csv')\n",
    "\n",
    "# warm vs. hyp\n",
    "w.h_deg <- read.csv('/project/pi_sarah_gignouxwolfsohn_uml_edu/julia/CE_2024/CE24_RNA-seq/analysis/diff_expression/phase1_v_phase1/deseq_res_files/DEGs/DEG_h_w.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ee6d95c1-ac7d-4fab-b99e-77896f4665b1",
   "metadata": {},
   "source": [
    "create a vector of entrez gene id (only numbers)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "6a41a671-2506-4763-9fa2-8fbbd188a543",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'111121183'</li><li>'111120021'</li><li>'111119724'</li><li>'111119726'</li><li>'111125414'</li><li>'111123780'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item '111121183'\n",
       "\\item '111120021'\n",
       "\\item '111119724'\n",
       "\\item '111119726'\n",
       "\\item '111125414'\n",
       "\\item '111123780'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. '111121183'\n",
       "2. '111120021'\n",
       "3. '111119724'\n",
       "4. '111119726'\n",
       "5. '111125414'\n",
       "6. '111123780'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"111121183\" \"111120021\" \"111119724\" \"111119726\" \"111125414\" \"111123780\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# remove leading LOC\n",
    "wc <- str_replace(w.c_deg$X, 'LOC', '')\n",
    "hc <- str_replace(h.c_deg$X, 'LOC', '')\n",
    "bc <- str_replace(b.c_deg$X, 'LOC', '')\n",
    "hb <- str_replace(h.b_deg$X, 'LOC', '')\n",
    "wb <- str_replace(w.b_deg$X, 'LOC', '')\n",
    "wh <- str_replace(w.h_deg$X, 'LOC', '')\n",
    "\n",
    "# check it worked\n",
    "head(hc)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "eaba110f-a9b0-405c-ae14-e3a593ff5a31",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'111116054'</li><li>'111126949'</li><li>'111110729'</li><li>'111112434'</li><li>'111120752'</li><li>'111128944'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item '111116054'\n",
       "\\item '111126949'\n",
       "\\item '111110729'\n",
       "\\item '111112434'\n",
       "\\item '111120752'\n",
       "\\item '111128944'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. '111116054'\n",
       "2. '111126949'\n",
       "3. '111110729'\n",
       "4. '111112434'\n",
       "5. '111120752'\n",
       "6. '111128944'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"111116054\" \"111126949\" \"111110729\" \"111112434\" \"111120752\" \"111128944\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "universe_list <- read.csv('/project/pi_sarah_gignouxwolfsohn_uml_edu/julia/CE_2024/CE24_RNA-seq/analysis/diff_expression/phase1_v_phase1/deseq_res_files/all_genes/p1.warm_v_cont.csv') \n",
    "universe_list <- str_replace(universe_list$X, 'LOC', '')\n",
    "head(universe_list)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6664bf87-c96e-4a90-9ca9-81b540c9c178",
   "metadata": {},
   "source": [
    "## 2. run `enrichKEGG`"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a168d39a-8627-4706-9597-250b85ebfe69",
   "metadata": {},
   "source": [
    "### Phase 1 Warm vs. Control"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "111d4f5e-f019-4c8f-b689-a9d30ef6bd85",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading KEGG annotation online: \"https://rest.kegg.jp/link/cvn/pathway\"...\n",
      "\n",
      "Reading KEGG annotation online: \"https://rest.kegg.jp/list/pathway/cvn\"...\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 11</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>category</th><th scope=col>subcategory</th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>cvn00910</th><td>Metabolism                    </td><td>Energy metabolism               </td><td>cvn00910</td><td>Nitrogen metabolism - Crassostrea virginica (eastern oyster)           </td><td>2/6</td><td>27/5527 </td><td>0.000340629</td><td>0.001362516</td><td>0.0007171138</td><td>111137424/111117514</td><td>2</td></tr>\n",
       "\t<tr><th scope=row>cvn04120</th><td>Genetic Information Processing</td><td>Folding, sorting and degradation</td><td>cvn04120</td><td>Ubiquitin mediated proteolysis - Crassostrea virginica (eastern oyster)</td><td>2/6</td><td>186/5527</td><td>0.015453701</td><td>0.030907402</td><td>0.0162670539</td><td>111123894/111130636</td><td>2</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 11\n",
       "\\begin{tabular}{r|lllllllllll}\n",
       "  & category & subcategory & ID & Description & GeneRatio & BgRatio & pvalue & p.adjust & qvalue & geneID & Count\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\tcvn00910 & Metabolism                     & Energy metabolism                & cvn00910 & Nitrogen metabolism - Crassostrea virginica (eastern oyster)            & 2/6 & 27/5527  & 0.000340629 & 0.001362516 & 0.0007171138 & 111137424/111117514 & 2\\\\\n",
       "\tcvn04120 & Genetic Information Processing & Folding, sorting and degradation & cvn04120 & Ubiquitin mediated proteolysis - Crassostrea virginica (eastern oyster) & 2/6 & 186/5527 & 0.015453701 & 0.030907402 & 0.0162670539 & 111123894/111130636 & 2\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 11\n",
       "\n",
       "| <!--/--> | category &lt;chr&gt; | subcategory &lt;chr&gt; | ID &lt;chr&gt; | Description &lt;chr&gt; | GeneRatio &lt;chr&gt; | BgRatio &lt;chr&gt; | pvalue &lt;dbl&gt; | p.adjust &lt;dbl&gt; | qvalue &lt;dbl&gt; | geneID &lt;chr&gt; | Count &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| cvn00910 | Metabolism                     | Energy metabolism                | cvn00910 | Nitrogen metabolism - Crassostrea virginica (eastern oyster)            | 2/6 | 27/5527  | 0.000340629 | 0.001362516 | 0.0007171138 | 111137424/111117514 | 2 |\n",
       "| cvn04120 | Genetic Information Processing | Folding, sorting and degradation | cvn04120 | Ubiquitin mediated proteolysis - Crassostrea virginica (eastern oyster) | 2/6 | 186/5527 | 0.015453701 | 0.030907402 | 0.0162670539 | 111123894/111130636 | 2 |\n",
       "\n"
      ],
      "text/plain": [
       "         category                       subcategory                     \n",
       "cvn00910 Metabolism                     Energy metabolism               \n",
       "cvn04120 Genetic Information Processing Folding, sorting and degradation\n",
       "         ID      \n",
       "cvn00910 cvn00910\n",
       "cvn04120 cvn04120\n",
       "         Description                                                            \n",
       "cvn00910 Nitrogen metabolism - Crassostrea virginica (eastern oyster)           \n",
       "cvn04120 Ubiquitin mediated proteolysis - Crassostrea virginica (eastern oyster)\n",
       "         GeneRatio BgRatio  pvalue      p.adjust    qvalue      \n",
       "cvn00910 2/6       27/5527  0.000340629 0.001362516 0.0007171138\n",
       "cvn04120 2/6       186/5527 0.015453701 0.030907402 0.0162670539\n",
       "         geneID              Count\n",
       "cvn00910 111137424/111117514 2    \n",
       "cvn04120 111123894/111130636 2    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "wc.kegg <- enrichKEGG(\n",
    "    gene = wc,\n",
    "    organism = 'cvn',\n",
    "    keyType = 'kegg',\n",
    "    pvalueCutoff = 0.05,\n",
    "    pAdjustMethod = 'BH',\n",
    "    universe = universe_list)\n",
    "\n",
    "head(wc.kegg)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c940054e-a39c-43cc-a426-4784f2253cff",
   "metadata": {},
   "source": [
    "### Phase 1 Hypoxic vs. Control"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "f97f787a-03dc-4a37-8af2-b75c0be4cb3c",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "--> No gene can be mapped....\n",
      "\n",
      "--> Expected input gene ID: 111127323,111138520,111134492,111109888,111111035,111109436\n",
      "\n",
      "--> return NULL...\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "NULL"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "hc.kegg <- enrichKEGG(\n",
    "    gene = hc,\n",
    "    organism = 'cvn',\n",
    "    keyType = 'kegg',\n",
    "    pvalueCutoff = 0.05,\n",
    "    pAdjustMethod = 'BH',\n",
    "    universe = universe_list)\n",
    "\n",
    "head(hc.kegg)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6308a50a-8dcc-4c70-86a7-0433c58fb01c",
   "metadata": {},
   "source": [
    "either all 53 DEGs are not annotated in the KEGG database, or none of the pathways are enriched - since below, there are some dfs that are empty, I think the prior hypothesis is true"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9b14e9fd-8c67-4ae9-a56a-716a378e7f10",
   "metadata": {},
   "source": [
    "### Phase 1 Both vs. Control"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "9d96db6f-6086-4d07-9f3a-8f02ac9585a8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 11</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>category</th><th scope=col>subcategory</th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>cvn00520</th><td>Metabolism        </td><td>Carbohydrate metabolism</td><td>cvn00520</td><td>Amino sugar and nucleotide sugar metabolism - Crassostrea virginica (eastern oyster)</td><td>6/24</td><td>75/5527</td><td>5.655051e-07</td><td>1.187561e-05</td><td>8.333759e-06</td><td>111119837/111124157/111125216/111125215/111127514/111125342</td><td>6</td></tr>\n",
       "\t<tr><th scope=row>cvn04977</th><td>Organismal Systems</td><td>Digestive system       </td><td>cvn04977</td><td>Vitamin digestion and absorption - Crassostrea virginica (eastern oyster)           </td><td>4/24</td><td>83/5527</td><td>3.996778e-04</td><td>4.196617e-03</td><td>2.944994e-03</td><td>111125053/111103328/111103330/111111573                    </td><td>4</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 11\n",
       "\\begin{tabular}{r|lllllllllll}\n",
       "  & category & subcategory & ID & Description & GeneRatio & BgRatio & pvalue & p.adjust & qvalue & geneID & Count\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\tcvn00520 & Metabolism         & Carbohydrate metabolism & cvn00520 & Amino sugar and nucleotide sugar metabolism - Crassostrea virginica (eastern oyster) & 6/24 & 75/5527 & 5.655051e-07 & 1.187561e-05 & 8.333759e-06 & 111119837/111124157/111125216/111125215/111127514/111125342 & 6\\\\\n",
       "\tcvn04977 & Organismal Systems & Digestive system        & cvn04977 & Vitamin digestion and absorption - Crassostrea virginica (eastern oyster)            & 4/24 & 83/5527 & 3.996778e-04 & 4.196617e-03 & 2.944994e-03 & 111125053/111103328/111103330/111111573                     & 4\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 11\n",
       "\n",
       "| <!--/--> | category &lt;chr&gt; | subcategory &lt;chr&gt; | ID &lt;chr&gt; | Description &lt;chr&gt; | GeneRatio &lt;chr&gt; | BgRatio &lt;chr&gt; | pvalue &lt;dbl&gt; | p.adjust &lt;dbl&gt; | qvalue &lt;dbl&gt; | geneID &lt;chr&gt; | Count &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| cvn00520 | Metabolism         | Carbohydrate metabolism | cvn00520 | Amino sugar and nucleotide sugar metabolism - Crassostrea virginica (eastern oyster) | 6/24 | 75/5527 | 5.655051e-07 | 1.187561e-05 | 8.333759e-06 | 111119837/111124157/111125216/111125215/111127514/111125342 | 6 |\n",
       "| cvn04977 | Organismal Systems | Digestive system        | cvn04977 | Vitamin digestion and absorption - Crassostrea virginica (eastern oyster)            | 4/24 | 83/5527 | 3.996778e-04 | 4.196617e-03 | 2.944994e-03 | 111125053/111103328/111103330/111111573                     | 4 |\n",
       "\n"
      ],
      "text/plain": [
       "         category           subcategory             ID      \n",
       "cvn00520 Metabolism         Carbohydrate metabolism cvn00520\n",
       "cvn04977 Organismal Systems Digestive system        cvn04977\n",
       "         Description                                                                         \n",
       "cvn00520 Amino sugar and nucleotide sugar metabolism - Crassostrea virginica (eastern oyster)\n",
       "cvn04977 Vitamin digestion and absorption - Crassostrea virginica (eastern oyster)           \n",
       "         GeneRatio BgRatio pvalue       p.adjust     qvalue      \n",
       "cvn00520 6/24      75/5527 5.655051e-07 1.187561e-05 8.333759e-06\n",
       "cvn04977 4/24      83/5527 3.996778e-04 4.196617e-03 2.944994e-03\n",
       "         geneID                                                      Count\n",
       "cvn00520 111119837/111124157/111125216/111125215/111127514/111125342 6    \n",
       "cvn04977 111125053/111103328/111103330/111111573                     4    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "bc.kegg <- enrichKEGG(\n",
    "    gene = bc,\n",
    "    organism = 'cvn',\n",
    "    keyType = 'kegg',\n",
    "    pvalueCutoff = 0.05,\n",
    "    pAdjustMethod = 'BH',\n",
    "    universe = universe_list)\n",
    "\n",
    "head(bc.kegg)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1786996a-46d8-43d7-b2b9-2d586ea137c6",
   "metadata": {},
   "source": [
    "### Phase 1 Hypoxic vs. Both"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "f49ae35a-5310-4b31-b69e-c0c52ff3c5b6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 1 × 11</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>category</th><th scope=col>subcategory</th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>cvn04977</th><td>Organismal Systems</td><td>Digestive system</td><td>cvn04977</td><td>Vitamin digestion and absorption - Crassostrea virginica (eastern oyster)</td><td>3/14</td><td>83/5527</td><td>0.001054943</td><td>0.0147692</td><td>0.01110466</td><td>111103328/111103330/111111573</td><td>3</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 1 × 11\n",
       "\\begin{tabular}{r|lllllllllll}\n",
       "  & category & subcategory & ID & Description & GeneRatio & BgRatio & pvalue & p.adjust & qvalue & geneID & Count\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\tcvn04977 & Organismal Systems & Digestive system & cvn04977 & Vitamin digestion and absorption - Crassostrea virginica (eastern oyster) & 3/14 & 83/5527 & 0.001054943 & 0.0147692 & 0.01110466 & 111103328/111103330/111111573 & 3\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 1 × 11\n",
       "\n",
       "| <!--/--> | category &lt;chr&gt; | subcategory &lt;chr&gt; | ID &lt;chr&gt; | Description &lt;chr&gt; | GeneRatio &lt;chr&gt; | BgRatio &lt;chr&gt; | pvalue &lt;dbl&gt; | p.adjust &lt;dbl&gt; | qvalue &lt;dbl&gt; | geneID &lt;chr&gt; | Count &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| cvn04977 | Organismal Systems | Digestive system | cvn04977 | Vitamin digestion and absorption - Crassostrea virginica (eastern oyster) | 3/14 | 83/5527 | 0.001054943 | 0.0147692 | 0.01110466 | 111103328/111103330/111111573 | 3 |\n",
       "\n"
      ],
      "text/plain": [
       "         category           subcategory      ID      \n",
       "cvn04977 Organismal Systems Digestive system cvn04977\n",
       "         Description                                                              \n",
       "cvn04977 Vitamin digestion and absorption - Crassostrea virginica (eastern oyster)\n",
       "         GeneRatio BgRatio pvalue      p.adjust  qvalue    \n",
       "cvn04977 3/14      83/5527 0.001054943 0.0147692 0.01110466\n",
       "         geneID                        Count\n",
       "cvn04977 111103328/111103330/111111573 3    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "hb.kegg <- enrichKEGG(\n",
    "    gene = hb,\n",
    "    organism = 'cvn',\n",
    "    keyType = 'kegg',\n",
    "    pvalueCutoff = 0.05,\n",
    "    pAdjustMethod = 'BH',\n",
    "    universe = universe_list)\n",
    "\n",
    "head(hb.kegg)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3f203902-3cfb-46a1-9101-ef51462ccc0a",
   "metadata": {},
   "source": [
    "### Phase 1 Warm vs. Both"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "c17b61ab-307a-4170-9265-25ebbf76fce4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 0 × 11</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>category</th><th scope=col>subcategory</th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 0 × 11\n",
       "\\begin{tabular}{lllllllllll}\n",
       " category & subcategory & ID & Description & GeneRatio & BgRatio & pvalue & p.adjust & qvalue & geneID & Count\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 0 × 11\n",
       "\n",
       "| category &lt;chr&gt; | subcategory &lt;chr&gt; | ID &lt;chr&gt; | Description &lt;chr&gt; | GeneRatio &lt;chr&gt; | BgRatio &lt;chr&gt; | pvalue &lt;dbl&gt; | p.adjust &lt;dbl&gt; | qvalue &lt;dbl&gt; | geneID &lt;chr&gt; | Count &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|\n",
       "\n"
      ],
      "text/plain": [
       "     category subcategory ID Description GeneRatio BgRatio pvalue p.adjust\n",
       "     qvalue geneID Count"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "wb.kegg <- enrichKEGG(\n",
    "    gene = wb,\n",
    "    organism = 'cvn',\n",
    "    keyType = 'kegg',\n",
    "    pvalueCutoff = 0.05,\n",
    "    pAdjustMethod = 'BH',\n",
    "    universe = universe_list)\n",
    "\n",
    "head(wb.kegg)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "987e7da1-b3a2-4b65-a38e-724bb22e478d",
   "metadata": {},
   "source": [
    "so no enriched KEGG pathways"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e66a3f2a-e74e-406c-8bf7-f0e5d2d7d1cb",
   "metadata": {},
   "source": [
    "### Phase 1 Warm vs. Hypoxic"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "8a137d91-db47-4b6e-bdb2-b9d0c1e47135",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 0 × 11</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>category</th><th scope=col>subcategory</th><th scope=col>ID</th><th scope=col>Description</th><th scope=col>GeneRatio</th><th scope=col>BgRatio</th><th scope=col>pvalue</th><th scope=col>p.adjust</th><th scope=col>qvalue</th><th scope=col>geneID</th><th scope=col>Count</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 0 × 11\n",
       "\\begin{tabular}{lllllllllll}\n",
       " category & subcategory & ID & Description & GeneRatio & BgRatio & pvalue & p.adjust & qvalue & geneID & Count\\\\\n",
       " <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <dbl> & <dbl> & <dbl> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 0 × 11\n",
       "\n",
       "| category &lt;chr&gt; | subcategory &lt;chr&gt; | ID &lt;chr&gt; | Description &lt;chr&gt; | GeneRatio &lt;chr&gt; | BgRatio &lt;chr&gt; | pvalue &lt;dbl&gt; | p.adjust &lt;dbl&gt; | qvalue &lt;dbl&gt; | geneID &lt;chr&gt; | Count &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|\n",
       "\n"
      ],
      "text/plain": [
       "     category subcategory ID Description GeneRatio BgRatio pvalue p.adjust\n",
       "     qvalue geneID Count"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "wh.kegg <- enrichKEGG(\n",
    "    gene = wh,\n",
    "    organism = 'cvn',\n",
    "    keyType = 'kegg',\n",
    "    pvalueCutoff = 0.05,\n",
    "    pAdjustMethod = 'BH',\n",
    "    universe = universe_list)\n",
    "\n",
    "head(wh.kegg)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "052ad82b-47eb-4ed4-a230-80529b9f2fb7",
   "metadata": {},
   "source": [
    "no enriched KEGG pathways"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R [conda env:.conda-r-env]",
   "language": "R",
   "name": "conda-env-.conda-r-env-r"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.3.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
