{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "56fc3767-a53b-4fe8-9379-296f4b503ff0",
   "metadata": {},
   "source": [
    "# DESeq1\n",
    "# Phase 1 vs. Phase 2\n",
    "\n",
    "focusing on pairwise comparisons of oysters that only experienced phase 1 to oysters that experienced phase 2 - **how does gene expression change after a second, later environment**\n",
    "\n",
    "## 0. load libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "f7a8ab29-aa84-4c7b-922f-8cda6c19e1ff",
   "metadata": {},
   "outputs": [],
   "source": [
    "library(tidyverse)\n",
    "library(DESeq2)\n",
    "library(EnhancedVolcano)\n",
    "library(vegan)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8cef49e7-565a-4474-bb65-bc7e9987614b",
   "metadata": {},
   "source": [
    "## 1. read counts matrix and meta data\n",
    "\n",
    "### counts matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "ac91b154-a9b8-4400-a7a3-25f830385d31",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 120</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>B1_B1_O01</th><th scope=col>B1_Nu_O03</th><th scope=col>B1_W5_O50</th><th scope=col>B2_B5_O51</th><th scope=col>B2_C4_O40</th><th scope=col>B2_Nu_O12</th><th scope=col>B3_B4_O41</th><th scope=col>B3_C3_O30</th><th scope=col>B3_C6_O66</th><th scope=col>B3_H4_O41</th><th scope=col>⋯</th><th scope=col>W5_C4_G45</th><th scope=col>W5_H4_G46</th><th scope=col>W5_W2_G22</th><th scope=col>W6_B3_G35</th><th scope=col>W6_B4_G48</th><th scope=col>W6_H6_G71</th><th scope=col>W6_Nu_G41</th><th scope=col>W6_Nu_G45</th><th scope=col>W6_W3_G36</th><th scope=col>W6_W4_G48</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>⋯</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>LOC111116054</th><td>  2</td><td>  1</td><td>  6</td><td> 10</td><td>  9</td><td> 16</td><td>  0</td><td> 10</td><td> 22</td><td>  9</td><td>⋯</td><td>  9</td><td>  2</td><td>  8</td><td>  5</td><td>  2</td><td> 19</td><td>  1</td><td>  2</td><td>  1</td><td>  0</td></tr>\n",
       "\t<tr><th scope=row>LOC111126949</th><td>885</td><td>652</td><td>477</td><td>654</td><td>586</td><td>523</td><td>392</td><td>357</td><td>496</td><td>540</td><td>⋯</td><td>407</td><td>740</td><td>707</td><td>406</td><td>418</td><td>424</td><td>492</td><td>330</td><td>281</td><td>599</td></tr>\n",
       "\t<tr><th scope=row>LOC111110729</th><td> 64</td><td>209</td><td> 93</td><td> 63</td><td>100</td><td>177</td><td> 76</td><td> 98</td><td>235</td><td>181</td><td>⋯</td><td> 70</td><td>149</td><td>121</td><td>115</td><td>126</td><td>108</td><td>118</td><td>115</td><td>127</td><td>213</td></tr>\n",
       "\t<tr><th scope=row>LOC111112434</th><td> 11</td><td>  7</td><td>  2</td><td>  0</td><td>  2</td><td>  2</td><td> 12</td><td> 15</td><td> 10</td><td> 36</td><td>⋯</td><td> 11</td><td>  6</td><td>  4</td><td>  0</td><td>  2</td><td> 16</td><td> 22</td><td>  0</td><td>  8</td><td>  0</td></tr>\n",
       "\t<tr><th scope=row>LOC111120752</th><td>360</td><td>586</td><td>336</td><td>426</td><td>351</td><td>417</td><td>236</td><td>278</td><td>761</td><td>373</td><td>⋯</td><td>359</td><td>345</td><td>438</td><td>278</td><td>287</td><td>416</td><td>621</td><td>251</td><td>333</td><td>430</td></tr>\n",
       "\t<tr><th scope=row>LOC111128944</th><td>  0</td><td>  0</td><td>  0</td><td>  0</td><td>  0</td><td>  0</td><td>  0</td><td>  0</td><td>  0</td><td>  0</td><td>⋯</td><td>  0</td><td>  0</td><td>  0</td><td>  0</td><td>  0</td><td>  0</td><td>  0</td><td>  1</td><td>  0</td><td>  0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 120\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & B1\\_B1\\_O01 & B1\\_Nu\\_O03 & B1\\_W5\\_O50 & B2\\_B5\\_O51 & B2\\_C4\\_O40 & B2\\_Nu\\_O12 & B3\\_B4\\_O41 & B3\\_C3\\_O30 & B3\\_C6\\_O66 & B3\\_H4\\_O41 & ⋯ & W5\\_C4\\_G45 & W5\\_H4\\_G46 & W5\\_W2\\_G22 & W6\\_B3\\_G35 & W6\\_B4\\_G48 & W6\\_H6\\_G71 & W6\\_Nu\\_G41 & W6\\_Nu\\_G45 & W6\\_W3\\_G36 & W6\\_W4\\_G48\\\\\n",
       "  & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & ⋯ & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\tLOC111116054 &   2 &   1 &   6 &  10 &   9 &  16 &   0 &  10 &  22 &   9 & ⋯ &   9 &   2 &   8 &   5 &   2 &  19 &   1 &   2 &   1 &   0\\\\\n",
       "\tLOC111126949 & 885 & 652 & 477 & 654 & 586 & 523 & 392 & 357 & 496 & 540 & ⋯ & 407 & 740 & 707 & 406 & 418 & 424 & 492 & 330 & 281 & 599\\\\\n",
       "\tLOC111110729 &  64 & 209 &  93 &  63 & 100 & 177 &  76 &  98 & 235 & 181 & ⋯ &  70 & 149 & 121 & 115 & 126 & 108 & 118 & 115 & 127 & 213\\\\\n",
       "\tLOC111112434 &  11 &   7 &   2 &   0 &   2 &   2 &  12 &  15 &  10 &  36 & ⋯ &  11 &   6 &   4 &   0 &   2 &  16 &  22 &   0 &   8 &   0\\\\\n",
       "\tLOC111120752 & 360 & 586 & 336 & 426 & 351 & 417 & 236 & 278 & 761 & 373 & ⋯ & 359 & 345 & 438 & 278 & 287 & 416 & 621 & 251 & 333 & 430\\\\\n",
       "\tLOC111128944 &   0 &   0 &   0 &   0 &   0 &   0 &   0 &   0 &   0 &   0 & ⋯ &   0 &   0 &   0 &   0 &   0 &   0 &   0 &   1 &   0 &   0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 120\n",
       "\n",
       "| <!--/--> | B1_B1_O01 &lt;int&gt; | B1_Nu_O03 &lt;int&gt; | B1_W5_O50 &lt;int&gt; | B2_B5_O51 &lt;int&gt; | B2_C4_O40 &lt;int&gt; | B2_Nu_O12 &lt;int&gt; | B3_B4_O41 &lt;int&gt; | B3_C3_O30 &lt;int&gt; | B3_C6_O66 &lt;int&gt; | B3_H4_O41 &lt;int&gt; | ⋯ ⋯ | W5_C4_G45 &lt;int&gt; | W5_H4_G46 &lt;int&gt; | W5_W2_G22 &lt;int&gt; | W6_B3_G35 &lt;int&gt; | W6_B4_G48 &lt;int&gt; | W6_H6_G71 &lt;int&gt; | W6_Nu_G41 &lt;int&gt; | W6_Nu_G45 &lt;int&gt; | W6_W3_G36 &lt;int&gt; | W6_W4_G48 &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| LOC111116054 |   2 |   1 |   6 |  10 |   9 |  16 |   0 |  10 |  22 |   9 | ⋯ |   9 |   2 |   8 |   5 |   2 |  19 |   1 |   2 |   1 |   0 |\n",
       "| LOC111126949 | 885 | 652 | 477 | 654 | 586 | 523 | 392 | 357 | 496 | 540 | ⋯ | 407 | 740 | 707 | 406 | 418 | 424 | 492 | 330 | 281 | 599 |\n",
       "| LOC111110729 |  64 | 209 |  93 |  63 | 100 | 177 |  76 |  98 | 235 | 181 | ⋯ |  70 | 149 | 121 | 115 | 126 | 108 | 118 | 115 | 127 | 213 |\n",
       "| LOC111112434 |  11 |   7 |   2 |   0 |   2 |   2 |  12 |  15 |  10 |  36 | ⋯ |  11 |   6 |   4 |   0 |   2 |  16 |  22 |   0 |   8 |   0 |\n",
       "| LOC111120752 | 360 | 586 | 336 | 426 | 351 | 417 | 236 | 278 | 761 | 373 | ⋯ | 359 | 345 | 438 | 278 | 287 | 416 | 621 | 251 | 333 | 430 |\n",
       "| LOC111128944 |   0 |   0 |   0 |   0 |   0 |   0 |   0 |   0 |   0 |   0 | ⋯ |   0 |   0 |   0 |   0 |   0 |   0 |   0 |   1 |   0 |   0 |\n",
       "\n"
      ],
      "text/plain": [
       "             B1_B1_O01 B1_Nu_O03 B1_W5_O50 B2_B5_O51 B2_C4_O40 B2_Nu_O12\n",
       "LOC111116054   2         1         6        10         9        16      \n",
       "LOC111126949 885       652       477       654       586       523      \n",
       "LOC111110729  64       209        93        63       100       177      \n",
       "LOC111112434  11         7         2         0         2         2      \n",
       "LOC111120752 360       586       336       426       351       417      \n",
       "LOC111128944   0         0         0         0         0         0      \n",
       "             B3_B4_O41 B3_C3_O30 B3_C6_O66 B3_H4_O41 ⋯ W5_C4_G45 W5_H4_G46\n",
       "LOC111116054   0        10        22         9       ⋯   9         2      \n",
       "LOC111126949 392       357       496       540       ⋯ 407       740      \n",
       "LOC111110729  76        98       235       181       ⋯  70       149      \n",
       "LOC111112434  12        15        10        36       ⋯  11         6      \n",
       "LOC111120752 236       278       761       373       ⋯ 359       345      \n",
       "LOC111128944   0         0         0         0       ⋯   0         0      \n",
       "             W5_W2_G22 W6_B3_G35 W6_B4_G48 W6_H6_G71 W6_Nu_G41 W6_Nu_G45\n",
       "LOC111116054   8         5         2        19         1         2      \n",
       "LOC111126949 707       406       418       424       492       330      \n",
       "LOC111110729 121       115       126       108       118       115      \n",
       "LOC111112434   4         0         2        16        22         0      \n",
       "LOC111120752 438       278       287       416       621       251      \n",
       "LOC111128944   0         0         0         0         0         1      \n",
       "             W6_W3_G36 W6_W4_G48\n",
       "LOC111116054   1         0      \n",
       "LOC111126949 281       599      \n",
       "LOC111110729 127       213      \n",
       "LOC111112434   8         0      \n",
       "LOC111120752 333       430      \n",
       "LOC111128944   0         0      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# counts matrix\n",
    "counts <- read.csv('/work/pi_sarah_gignouxwolfsohn_uml_edu/julia_mcdonough_student_uml_edu/ce24_rnaseq/featureCounts/featureCounts_matrix.csv')\n",
    "\n",
    "# set gene_ID as row name\n",
    "rownames(counts) <- counts$Gene_ID\n",
    "\n",
    "# remove gene id and length\n",
    "counts.df <- counts %>%\n",
    "select(-Gene_ID, -Length)\n",
    "\n",
    "head(counts.df)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f4fad8e3-b740-4181-8386-292e2a00a8dd",
   "metadata": {},
   "source": [
    "### meta data\n",
    "the meta data csv was already generated in [deseq_p1.v.p1](https://github.com/jgmcdonough/CE24_RNA-seq/blob/main/analysis/diff_expression/phase1_v_phase1/deseq_p1.v.p1.ipynb), so just reading in here"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "1a05939d-9d6c-48d5-9641-6ddaea54842a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 8</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Phase1_treatment</th><th scope=col>Phase1_temp</th><th scope=col>Phase1_DO</th><th scope=col>Phase1_TankRep</th><th scope=col>Phase2_treatment</th><th scope=col>Phase2_temp</th><th scope=col>Phase2_DO</th><th scope=col>Phase2_TankRep</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>B1_B1_O01</th><td>both</td><td>warm</td><td>hypoxic</td><td>1</td><td>both   </td><td>warm   </td><td>hypoxic </td><td> 1</td></tr>\n",
       "\t<tr><th scope=row>B1_Nu_O03</th><td>both</td><td>warm</td><td>hypoxic</td><td>1</td><td>NA     </td><td>NA     </td><td>NA      </td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>B1_W5_O50</th><td>both</td><td>warm</td><td>hypoxic</td><td>1</td><td>warm   </td><td>warm   </td><td>normoxic</td><td> 5</td></tr>\n",
       "\t<tr><th scope=row>B2_B5_O51</th><td>both</td><td>warm</td><td>hypoxic</td><td>2</td><td>both   </td><td>warm   </td><td>hypoxic </td><td> 5</td></tr>\n",
       "\t<tr><th scope=row>B2_C4_O40</th><td>both</td><td>warm</td><td>hypoxic</td><td>2</td><td>control</td><td>ambient</td><td>normoxic</td><td> 4</td></tr>\n",
       "\t<tr><th scope=row>B2_Nu_O12</th><td>both</td><td>warm</td><td>hypoxic</td><td>2</td><td>NA     </td><td>NA     </td><td>NA      </td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 8\n",
       "\\begin{tabular}{r|llllllll}\n",
       "  & Phase1\\_treatment & Phase1\\_temp & Phase1\\_DO & Phase1\\_TankRep & Phase2\\_treatment & Phase2\\_temp & Phase2\\_DO & Phase2\\_TankRep\\\\\n",
       "  & <chr> & <chr> & <chr> & <int> & <chr> & <chr> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\tB1\\_B1\\_O01 & both & warm & hypoxic & 1 & both    & warm    & hypoxic  &  1\\\\\n",
       "\tB1\\_Nu\\_O03 & both & warm & hypoxic & 1 & NA      & NA      & NA       & NA\\\\\n",
       "\tB1\\_W5\\_O50 & both & warm & hypoxic & 1 & warm    & warm    & normoxic &  5\\\\\n",
       "\tB2\\_B5\\_O51 & both & warm & hypoxic & 2 & both    & warm    & hypoxic  &  5\\\\\n",
       "\tB2\\_C4\\_O40 & both & warm & hypoxic & 2 & control & ambient & normoxic &  4\\\\\n",
       "\tB2\\_Nu\\_O12 & both & warm & hypoxic & 2 & NA      & NA      & NA       & NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 8\n",
       "\n",
       "| <!--/--> | Phase1_treatment &lt;chr&gt; | Phase1_temp &lt;chr&gt; | Phase1_DO &lt;chr&gt; | Phase1_TankRep &lt;int&gt; | Phase2_treatment &lt;chr&gt; | Phase2_temp &lt;chr&gt; | Phase2_DO &lt;chr&gt; | Phase2_TankRep &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|\n",
       "| B1_B1_O01 | both | warm | hypoxic | 1 | both    | warm    | hypoxic  |  1 |\n",
       "| B1_Nu_O03 | both | warm | hypoxic | 1 | NA      | NA      | NA       | NA |\n",
       "| B1_W5_O50 | both | warm | hypoxic | 1 | warm    | warm    | normoxic |  5 |\n",
       "| B2_B5_O51 | both | warm | hypoxic | 2 | both    | warm    | hypoxic  |  5 |\n",
       "| B2_C4_O40 | both | warm | hypoxic | 2 | control | ambient | normoxic |  4 |\n",
       "| B2_Nu_O12 | both | warm | hypoxic | 2 | NA      | NA      | NA       | NA |\n",
       "\n"
      ],
      "text/plain": [
       "          Phase1_treatment Phase1_temp Phase1_DO Phase1_TankRep\n",
       "B1_B1_O01 both             warm        hypoxic   1             \n",
       "B1_Nu_O03 both             warm        hypoxic   1             \n",
       "B1_W5_O50 both             warm        hypoxic   1             \n",
       "B2_B5_O51 both             warm        hypoxic   2             \n",
       "B2_C4_O40 both             warm        hypoxic   2             \n",
       "B2_Nu_O12 both             warm        hypoxic   2             \n",
       "          Phase2_treatment Phase2_temp Phase2_DO Phase2_TankRep\n",
       "B1_B1_O01 both             warm        hypoxic    1            \n",
       "B1_Nu_O03 NA               NA          NA        NA            \n",
       "B1_W5_O50 warm             warm        normoxic   5            \n",
       "B2_B5_O51 both             warm        hypoxic    5            \n",
       "B2_C4_O40 control          ambient     normoxic   4            \n",
       "B2_Nu_O12 NA               NA          NA        NA            "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "meta <- read.csv('/project/pi_sarah_gignouxwolfsohn_uml_edu/julia/CE_2024/CE24_RNA-seq/sample_metaData.csv') %>%\n",
    "filter(Sample != 'B3_Nu_O24') # remove failed QC sample\n",
    "\n",
    "# setting samples as rownames and deleting col from df\n",
    "rownames(meta) <- meta$Sample\n",
    "meta <- meta[, !colnames(meta) %in% \"Sample\"]\n",
    "\n",
    "head(meta)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a177ac4c-7c41-4e8e-9d9c-55182fd2baf8",
   "metadata": {},
   "source": [
    "creating new column for 'complete' treatment - so I can easily compare phase 1 to phase 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "a8d26501-663b-4f0d-88ca-c9bd3a69439d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 9</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Phase1_treatment</th><th scope=col>Phase1_temp</th><th scope=col>Phase1_DO</th><th scope=col>Phase1_TankRep</th><th scope=col>Phase2_treatment</th><th scope=col>Phase2_temp</th><th scope=col>Phase2_DO</th><th scope=col>Phase2_TankRep</th><th scope=col>complete_trtmt</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>B1_B1_O01</th><td>both</td><td>warm</td><td>hypoxic</td><td>1</td><td>both</td><td>warm</td><td>hypoxic</td><td> 1</td><td>both both</td></tr>\n",
       "\t<tr><th scope=row>B1_Nu_O03</th><td>both</td><td>warm</td><td>hypoxic</td><td>1</td><td>NA  </td><td>NA  </td><td>NA     </td><td>NA</td><td>both NA  </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 9\n",
       "\\begin{tabular}{r|lllllllll}\n",
       "  & Phase1\\_treatment & Phase1\\_temp & Phase1\\_DO & Phase1\\_TankRep & Phase2\\_treatment & Phase2\\_temp & Phase2\\_DO & Phase2\\_TankRep & complete\\_trtmt\\\\\n",
       "  & <chr> & <chr> & <chr> & <int> & <chr> & <chr> & <chr> & <int> & <chr>\\\\\n",
       "\\hline\n",
       "\tB1\\_B1\\_O01 & both & warm & hypoxic & 1 & both & warm & hypoxic &  1 & both both\\\\\n",
       "\tB1\\_Nu\\_O03 & both & warm & hypoxic & 1 & NA   & NA   & NA      & NA & both NA  \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 9\n",
       "\n",
       "| <!--/--> | Phase1_treatment &lt;chr&gt; | Phase1_temp &lt;chr&gt; | Phase1_DO &lt;chr&gt; | Phase1_TankRep &lt;int&gt; | Phase2_treatment &lt;chr&gt; | Phase2_temp &lt;chr&gt; | Phase2_DO &lt;chr&gt; | Phase2_TankRep &lt;int&gt; | complete_trtmt &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|\n",
       "| B1_B1_O01 | both | warm | hypoxic | 1 | both | warm | hypoxic |  1 | both both |\n",
       "| B1_Nu_O03 | both | warm | hypoxic | 1 | NA   | NA   | NA      | NA | both NA   |\n",
       "\n"
      ],
      "text/plain": [
       "          Phase1_treatment Phase1_temp Phase1_DO Phase1_TankRep\n",
       "B1_B1_O01 both             warm        hypoxic   1             \n",
       "B1_Nu_O03 both             warm        hypoxic   1             \n",
       "          Phase2_treatment Phase2_temp Phase2_DO Phase2_TankRep complete_trtmt\n",
       "B1_B1_O01 both             warm        hypoxic    1             both both     \n",
       "B1_Nu_O03 NA               NA          NA        NA             both NA       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 2 × 9</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Phase1_treatment</th><th scope=col>Phase1_temp</th><th scope=col>Phase1_DO</th><th scope=col>Phase1_TankRep</th><th scope=col>Phase2_treatment</th><th scope=col>Phase2_temp</th><th scope=col>Phase2_DO</th><th scope=col>Phase2_TankRep</th><th scope=col>complete_trtmt</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>W6_W3_G36</th><td>warm</td><td>warm</td><td>normoxic</td><td>6</td><td>warm</td><td>warm</td><td>normoxic</td><td>3</td><td>warm warm</td></tr>\n",
       "\t<tr><th scope=row>W6_W4_G48</th><td>warm</td><td>warm</td><td>normoxic</td><td>6</td><td>warm</td><td>warm</td><td>normoxic</td><td>4</td><td>warm warm</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 2 × 9\n",
       "\\begin{tabular}{r|lllllllll}\n",
       "  & Phase1\\_treatment & Phase1\\_temp & Phase1\\_DO & Phase1\\_TankRep & Phase2\\_treatment & Phase2\\_temp & Phase2\\_DO & Phase2\\_TankRep & complete\\_trtmt\\\\\n",
       "  & <chr> & <chr> & <chr> & <int> & <chr> & <chr> & <chr> & <int> & <chr>\\\\\n",
       "\\hline\n",
       "\tW6\\_W3\\_G36 & warm & warm & normoxic & 6 & warm & warm & normoxic & 3 & warm warm\\\\\n",
       "\tW6\\_W4\\_G48 & warm & warm & normoxic & 6 & warm & warm & normoxic & 4 & warm warm\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 2 × 9\n",
       "\n",
       "| <!--/--> | Phase1_treatment &lt;chr&gt; | Phase1_temp &lt;chr&gt; | Phase1_DO &lt;chr&gt; | Phase1_TankRep &lt;int&gt; | Phase2_treatment &lt;chr&gt; | Phase2_temp &lt;chr&gt; | Phase2_DO &lt;chr&gt; | Phase2_TankRep &lt;int&gt; | complete_trtmt &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|\n",
       "| W6_W3_G36 | warm | warm | normoxic | 6 | warm | warm | normoxic | 3 | warm warm |\n",
       "| W6_W4_G48 | warm | warm | normoxic | 6 | warm | warm | normoxic | 4 | warm warm |\n",
       "\n"
      ],
      "text/plain": [
       "          Phase1_treatment Phase1_temp Phase1_DO Phase1_TankRep\n",
       "W6_W3_G36 warm             warm        normoxic  6             \n",
       "W6_W4_G48 warm             warm        normoxic  6             \n",
       "          Phase2_treatment Phase2_temp Phase2_DO Phase2_TankRep complete_trtmt\n",
       "W6_W3_G36 warm             warm        normoxic  3              warm warm     \n",
       "W6_W4_G48 warm             warm        normoxic  4              warm warm     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "meta$complete_trtmt <- paste(meta$Phase1_treatment, meta$Phase2_treatment)\n",
    "\n",
    "# check\n",
    "head(meta,2)\n",
    "tail(meta,2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "68c83986-b1b5-4df8-9595-2c52e1d2b5ac",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "TRUE"
      ],
      "text/latex": [
       "TRUE"
      ],
      "text/markdown": [
       "TRUE"
      ],
      "text/plain": [
       "[1] TRUE"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# setting the same order for counts matrix and meta data\n",
    "counts.df <- counts.df[,rownames(meta)]\n",
    "all(rownames(meta) == colnames(counts.df))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "02143c3f-3d04-41b8-896a-8bc0ff44da30",
   "metadata": {},
   "source": [
    "## 2. DESeq2\n",
    "analyzing differential gene expression with [`DESeq2`](https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html)\n",
    "\n",
    "#### contrasts\n",
    "setting up contrasts (telling DESeq what is the control treatment)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "15ec2521-d584-4fc5-aaf9-5aa41cbb39ce",
   "metadata": {},
   "outputs": [],
   "source": [
    "# tempature treatments\n",
    "meta$Phase1_temp <- factor(meta$Phase1_temp,\n",
    "                                   levels = c('ambient', 'warm'))\n",
    "\n",
    "meta$Phase2_temp <- factor(meta$Phase2_temp,\n",
    "                                   levels = c('ambient', 'warm'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "ca0d38ba-3a49-4935-8988-99c2fa69334f",
   "metadata": {},
   "outputs": [],
   "source": [
    "# DO treatments\n",
    "meta$Phase1_DO <- factor(meta$Phase1_DO,\n",
    "                                   levels = c('normoxic', 'hypoxic'))\n",
    "\n",
    "meta$Phase2_DO <- factor(meta$Phase2_DO,\n",
    "                                   levels = c('normoxic', 'hypoxic'))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c8fb4317-1eda-45c1-9253-4dc6cc465ecd",
   "metadata": {},
   "source": [
    "### phase 1 overall treatments\n",
    "starting simple with looking at phase 1 treatments (not splitting into temperature/DO, just looking at combination of treatments)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "d39e354e-b04b-427e-97f8-b78b78503520",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in DESeqDataSet(se, design = design, ignoreRank):\n",
      "“some variables in design formula are characters, converting to factors”\n",
      "  Note: levels of factors in the design contain characters other than\n",
      "  letters, numbers, '_' and '.'. It is recommended (but not required) to use\n",
      "  only letters, numbers, and delimiters '_' or '.', as these are safe characters\n",
      "  for column names in R. [This is a message, not a warning or an error]\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "class: DESeqDataSet \n",
       "dim: 38838 119 \n",
       "metadata(1): version\n",
       "assays(1): counts\n",
       "rownames(38838): LOC111116054 LOC111126949 ... ND1 ND4L\n",
       "rowData names(0):\n",
       "colnames(119): B1_B1_O01 B1_Nu_O03 ... W6_W3_G36 W6_W4_G48\n",
       "colData names(9): Phase1_treatment Phase1_temp ... Phase2_TankRep\n",
       "  complete_trtmt"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "estimating size factors\n",
      "\n",
      "  Note: levels of factors in the design contain characters other than\n",
      "  letters, numbers, '_' and '.'. It is recommended (but not required) to use\n",
      "  only letters, numbers, and delimiters '_' or '.', as these are safe characters\n",
      "  for column names in R. [This is a message, not a warning or an error]\n",
      "\n",
      "estimating dispersions\n",
      "\n",
      "gene-wise dispersion estimates\n",
      "\n",
      "mean-dispersion relationship\n",
      "\n",
      "-- note: fitType='parametric', but the dispersion trend was not well captured by the\n",
      "   function: y = a/x + b, and a local regression fit was automatically substituted.\n",
      "   specify fitType='local' or 'mean' to avoid this message next time.\n",
      "\n",
      "  Note: levels of factors in the design contain characters other than\n",
      "  letters, numbers, '_' and '.'. It is recommended (but not required) to use\n",
      "  only letters, numbers, and delimiters '_' or '.', as these are safe characters\n",
      "  for column names in R. [This is a message, not a warning or an error]\n",
      "\n",
      "final dispersion estimates\n",
      "\n",
      "fitting model and testing\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'Intercept'</li><li>'complete_trtmt_both.control_vs_both.both'</li><li>'complete_trtmt_both.hypoxic_vs_both.both'</li><li>'complete_trtmt_both.NA_vs_both.both'</li><li>'complete_trtmt_both.warm_vs_both.both'</li><li>'complete_trtmt_control.both_vs_both.both'</li><li>'complete_trtmt_control.control_vs_both.both'</li><li>'complete_trtmt_control.hypoxic_vs_both.both'</li><li>'complete_trtmt_control.NA_vs_both.both'</li><li>'complete_trtmt_control.warm_vs_both.both'</li><li>'complete_trtmt_hypoxic.both_vs_both.both'</li><li>'complete_trtmt_hypoxic.control_vs_both.both'</li><li>'complete_trtmt_hypoxic.hypoxic_vs_both.both'</li><li>'complete_trtmt_hypoxic.NA_vs_both.both'</li><li>'complete_trtmt_hypoxic.warm_vs_both.both'</li><li>'complete_trtmt_warm.both_vs_both.both'</li><li>'complete_trtmt_warm.control_vs_both.both'</li><li>'complete_trtmt_warm.hypoxic_vs_both.both'</li><li>'complete_trtmt_warm.NA_vs_both.both'</li><li>'complete_trtmt_warm.warm_vs_both.both'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'Intercept'\n",
       "\\item 'complete\\_trtmt\\_both.control\\_vs\\_both.both'\n",
       "\\item 'complete\\_trtmt\\_both.hypoxic\\_vs\\_both.both'\n",
       "\\item 'complete\\_trtmt\\_both.NA\\_vs\\_both.both'\n",
       "\\item 'complete\\_trtmt\\_both.warm\\_vs\\_both.both'\n",
       "\\item 'complete\\_trtmt\\_control.both\\_vs\\_both.both'\n",
       "\\item 'complete\\_trtmt\\_control.control\\_vs\\_both.both'\n",
       "\\item 'complete\\_trtmt\\_control.hypoxic\\_vs\\_both.both'\n",
       "\\item 'complete\\_trtmt\\_control.NA\\_vs\\_both.both'\n",
       "\\item 'complete\\_trtmt\\_control.warm\\_vs\\_both.both'\n",
       "\\item 'complete\\_trtmt\\_hypoxic.both\\_vs\\_both.both'\n",
       "\\item 'complete\\_trtmt\\_hypoxic.control\\_vs\\_both.both'\n",
       "\\item 'complete\\_trtmt\\_hypoxic.hypoxic\\_vs\\_both.both'\n",
       "\\item 'complete\\_trtmt\\_hypoxic.NA\\_vs\\_both.both'\n",
       "\\item 'complete\\_trtmt\\_hypoxic.warm\\_vs\\_both.both'\n",
       "\\item 'complete\\_trtmt\\_warm.both\\_vs\\_both.both'\n",
       "\\item 'complete\\_trtmt\\_warm.control\\_vs\\_both.both'\n",
       "\\item 'complete\\_trtmt\\_warm.hypoxic\\_vs\\_both.both'\n",
       "\\item 'complete\\_trtmt\\_warm.NA\\_vs\\_both.both'\n",
       "\\item 'complete\\_trtmt\\_warm.warm\\_vs\\_both.both'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'Intercept'\n",
       "2. 'complete_trtmt_both.control_vs_both.both'\n",
       "3. 'complete_trtmt_both.hypoxic_vs_both.both'\n",
       "4. 'complete_trtmt_both.NA_vs_both.both'\n",
       "5. 'complete_trtmt_both.warm_vs_both.both'\n",
       "6. 'complete_trtmt_control.both_vs_both.both'\n",
       "7. 'complete_trtmt_control.control_vs_both.both'\n",
       "8. 'complete_trtmt_control.hypoxic_vs_both.both'\n",
       "9. 'complete_trtmt_control.NA_vs_both.both'\n",
       "10. 'complete_trtmt_control.warm_vs_both.both'\n",
       "11. 'complete_trtmt_hypoxic.both_vs_both.both'\n",
       "12. 'complete_trtmt_hypoxic.control_vs_both.both'\n",
       "13. 'complete_trtmt_hypoxic.hypoxic_vs_both.both'\n",
       "14. 'complete_trtmt_hypoxic.NA_vs_both.both'\n",
       "15. 'complete_trtmt_hypoxic.warm_vs_both.both'\n",
       "16. 'complete_trtmt_warm.both_vs_both.both'\n",
       "17. 'complete_trtmt_warm.control_vs_both.both'\n",
       "18. 'complete_trtmt_warm.hypoxic_vs_both.both'\n",
       "19. 'complete_trtmt_warm.NA_vs_both.both'\n",
       "20. 'complete_trtmt_warm.warm_vs_both.both'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"Intercept\"                                  \n",
       " [2] \"complete_trtmt_both.control_vs_both.both\"   \n",
       " [3] \"complete_trtmt_both.hypoxic_vs_both.both\"   \n",
       " [4] \"complete_trtmt_both.NA_vs_both.both\"        \n",
       " [5] \"complete_trtmt_both.warm_vs_both.both\"      \n",
       " [6] \"complete_trtmt_control.both_vs_both.both\"   \n",
       " [7] \"complete_trtmt_control.control_vs_both.both\"\n",
       " [8] \"complete_trtmt_control.hypoxic_vs_both.both\"\n",
       " [9] \"complete_trtmt_control.NA_vs_both.both\"     \n",
       "[10] \"complete_trtmt_control.warm_vs_both.both\"   \n",
       "[11] \"complete_trtmt_hypoxic.both_vs_both.both\"   \n",
       "[12] \"complete_trtmt_hypoxic.control_vs_both.both\"\n",
       "[13] \"complete_trtmt_hypoxic.hypoxic_vs_both.both\"\n",
       "[14] \"complete_trtmt_hypoxic.NA_vs_both.both\"     \n",
       "[15] \"complete_trtmt_hypoxic.warm_vs_both.both\"   \n",
       "[16] \"complete_trtmt_warm.both_vs_both.both\"      \n",
       "[17] \"complete_trtmt_warm.control_vs_both.both\"   \n",
       "[18] \"complete_trtmt_warm.hypoxic_vs_both.both\"   \n",
       "[19] \"complete_trtmt_warm.NA_vs_both.both\"        \n",
       "[20] \"complete_trtmt_warm.warm_vs_both.both\"      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# creating DESeq2 object\n",
    "dds <- DESeqDataSetFromMatrix(countData = counts.df,\n",
    "                              colData = meta,\n",
    "                              design = ~ complete_trtmt)\n",
    "dds\n",
    "\n",
    "# running DESeq\n",
    "dds <- DESeq(dds)\n",
    "\n",
    "# list coefficients\n",
    "resultsNames(dds)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R [conda env:.conda-r-env]",
   "language": "R",
   "name": "conda-env-.conda-r-env-r"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.3.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
